From: olh@suse.de
Subject: 2.6.11-rc2-bk6

 -- www.kernel.org/pub/linux/kernel/v2.6/snapshots/patch-2.6.11-rc2-bk5.log	2005-01-27 13:47:12.000000000 +0100
 ++ www.kernel.org/pub/linux/kernel/v2.6/snapshots/patch-2.6.11-rc2-bk6.log	2005-01-28 13:51:26.000000000 +0100
 ChangeSet@1.2140, 2005-01-26 10:43:01-08:00, herbert@13thfloor.at
 ChangeSet@1.1982, 2005-01-27 17:10:18-08:00, torvalds@ppc970.osdl.org
   Fix permissions on drivers/scsi/a100u2w.c
 
 ChangeSet@1.1966.85.6, 2005-01-28 00:08:57+00:00, dvrabel@com.rmk.(none)
   [ARM PATCH] 2437/1: ixp4xx: cosmetic change to arch_reset()
   
   Patch from David Vrabel
   
   Use some #define'd constants in the ixp4xx's arch_reset().
   
   Signed-off-by: David Vrabel
   Signed-off-by: Russell King
 
 ChangeSet@1.1966.86.5, 2005-01-27 17:13:18-05:00, shemminger@osdl.org
   [PATCH] (2/2) skfddi: netdev_priv and cast cleanup
   
   Use netdev_priv() and get rid of the PRIV() macro by using the correct
   data element. Also eliminate casts, where type is correct.
   
   Signed-off-by: Stephen Hemminger <shemminger@osdl.org>
   Signed-off-by: Jeff Garzik <jgarzik@pobox.com>
 
 ChangeSet@1.1966.86.4, 2005-01-27 17:12:59-05:00, shemminger@osdl.org
   [PATCH] (1/2) skfddi: initialization
   
   When the skfddi driver was converted to the new PCI netdevice interface,
   it never got tested with real hardware. The initialization got broken, this
   should fix it.
   
   Signed-off-by: Stephen Hemminger <shemminger@osdl.org>
   Signed-off-by: Jeff Garzik <jgarzik@pobox.com>
 
 ChangeSet@1.1966.86.3, 2005-01-27 17:08:40-05:00, galak@freescale.com
   [PATCH] netdrv gianfar: Fix usage of gfar_read in debug code
   
   Fixes instances where gfar_read() was invoked in debug codewith a value,
   rather than a pointer.
   
   Signed-of-by: Andy Fleming <afleming@freescale.com>
   Signed-of-by: Kumar Gala <afleming@freescale.com>
   Signed-off-by: Jeff Garzik <jgarzik@pobox.com>
 
 ChangeSet@1.1966.86.2, 2005-01-27 16:57:11-05:00, akpm@osdl.org
   [PATCH] e100 locking up netconsole.
   
   From: Steven Rostedt <rostedt@goodmis.org>
   
   If the transmit buffer of the e100 overflowed, then the system would hang.
   This was caused because the e100 driver would stop the queue, and
   netpoll_send_pkt in netpoll.c would then loop forever.  This is because the
   e100 net_poll would never start the queue again after the transmits have
   completed.
   
   For those that use the e100 and netconsole, all you need to do is a sysreq
   't' to lock up the system.
   
   Acked-by: Ingo Molnar <mingo@elte.hu>
   Signed-off-by: Andrew Morton <akpm@osdl.org>
   Signed-off-by: Jeff Garzik <jgarzik@pobox.com>
 
 ChangeSet@1.1966.1.176, 2005-01-27 16:52:54-05:00, jason.d.gaston@intel.com
   [PATCH] SATA AHCI support for Intel ICH7R
   
   This patch adds the Intel ICH7R DID's to the ahci.c SATA AHCI driver
   for ICH7R SATA support.
   
   Signed-off-by:  Jason Gaston <Jason.d.gaston@intel.com>
   Signed-off-by: Jeff Garzik <jgarzik@pobox.com>
 
 ChangeSet@1.1966.86.1, 2005-01-27 16:13:14-05:00, viro@parcelfarce.linux.theplanet.co.uk
   [PATCH] de2104x: Fixes breakage in getting MAC address
   
   Fixes breakage in getting MAC address on built-in tulip in
   alphastation 200 - without delay we are skipping bytes.
   
   Signed-off-by: Al Viro <viro@parcelfarce.linux.theplanet.co.uk>
   Signed-off-by: Jeff Garzik <jgarzik@pobox.com>
 
 ChangeSet@1.1966.1.175, 2005-01-27 15:37:39-05:00, haroldo.gamal@infolink.com.br
   [libata sata_sil] add another Seagate driver to blacklist
 
 ChangeSet@1.1966.85.5, 2005-01-27 19:31:47+00:00, rmk@flint.arm.linux.org.uk
   [ARM] Replace duplicate sets of vector code with assembler macro.
   
   Signed-off-by: Russell King <rmk@arm.linux.org.uk>
 
 ChangeSet@1.1966.85.4, 2005-01-27 19:13:38+00:00, rmk@flint.arm.linux.org.uk
   [ARM] Make vector labels consistent with naming scheme
   
   Signed-off-by: Russell King <rmk@arm.linux.org.uk>
 
 ChangeSet@1.1978, 2005-01-27 09:37:23-08:00, torvalds@ppc970.osdl.org
   Merge bk://kernel.bkbits.net/vojtech/for-linus
   into ppc970.osdl.org:/home/torvalds/v2.6/linux
 
 ChangeSet@1.1975.1.3, 2005-01-27 14:58:43+01:00, vojtech@silver.ucw.cz
   input: Add missing input_sync() calls to atkbd.c.
   
   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
 
 ChangeSet@1.1975.1.2, 2005-01-27 14:48:48+01:00, vojtech@silver.ucw.cz
   input: Enable scancode event generation in the HID driver. This should allow
          changing HID->event mappings (via EVIOCS*) in the future and make 
          debugging easier now.
   
   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
 
 ChangeSet@1.1975.1.1, 2005-01-27 13:47:19+01:00, vojtech@silver.ucw.cz
   input: Always bring the i8042 multiplexer out of multiplexing mode before
          rebooting.
   
   Signed-off-by: Vojtech Pavlik <vojtech@suse.cz>
 
 ChangeSet@1.1976, 2005-01-27 01:18:50-05:00, dtor_core@ameritech.net
   Input: ACK/NAK processing rules in libps2 were too strict - while it is a
          good idea to discard any character other than ACK/NAK during probe
          it causes missing releases and keys getting "stuck" when a command
          issued on enabled device. The effect is easily demonstrated with
          the following command:
              while true; do xset led 3; xset -led 3; done
   
          With this change extra characters will be discarded only if device
          has not been marked as "enabled" yet.
   
   Signed-off-by: Dmitry Torokhov <dtor@mail.ru> 
 
 ChangeSet@1.1966.1.174, 2005-01-26 10:43:01-08:00, herbert@13thfloor.at
 ChangeSet@1.2139, 2005-01-26 08:47:14-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.1.173, 2005-01-26 08:47:14-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.2020.1.8, 2005-01-25 22:06:30-08:00, tgraf@suug.ch
 ChangeSet@1.1966.81.8, 2005-01-25 22:06:30-08:00, tgraf@suug.ch
 ChangeSet@1.2138, 2005-01-25 22:00:23-08:00, James.Bottomley@SteelEye.com
 ChangeSet@1.1966.1.172, 2005-01-25 22:00:23-08:00, James.Bottomley@SteelEye.com
 ChangeSet@1.2020.1.7, 2005-01-25 21:53:42-08:00, herbert@gondor.apana.org.au
 ChangeSet@1.1966.81.7, 2005-01-25 21:53:42-08:00, herbert@gondor.apana.org.au
 ChangeSet@1.2020.2.18, 2005-01-25 21:34:49-08:00, akpm@osdl.org
 ChangeSet@1.1966.82.18, 2005-01-25 21:34:49-08:00, akpm@osdl.org
 ChangeSet@1.2020.2.17, 2005-01-25 21:33:24-08:00, akpm@osdl.org
 ChangeSet@1.1966.82.17, 2005-01-25 21:33:24-08:00, akpm@osdl.org
 ChangeSet@1.2020.2.16, 2005-01-25 21:31:12-08:00, mostrows@watson.ibm.com
 ChangeSet@1.1966.82.16, 2005-01-25 21:31:12-08:00, mostrows@watson.ibm.com
 ChangeSet@1.2020.2.15, 2005-01-25 21:19:37-08:00, rusty@rustcorp.com.au
 ChangeSet@1.1966.82.15, 2005-01-25 21:19:37-08:00, rusty@rustcorp.com.au
 ChangeSet@1.2020.2.14, 2005-01-25 21:18:13-08:00, rusty@rustcorp.com.au
 ChangeSet@1.1966.82.14, 2005-01-25 21:18:13-08:00, rusty@rustcorp.com.au
 ChangeSet@1.2020.2.13, 2005-01-25 21:16:24-08:00, rddunlap@osdl.org
 ChangeSet@1.1966.82.13, 2005-01-25 21:16:24-08:00, rddunlap@osdl.org
 ChangeSet@1.2137, 2005-01-25 21:15:28-08:00, nathanl@austin.ibm.com
 ChangeSet@1.1966.1.171, 2005-01-25 21:15:28-08:00, nathanl@austin.ibm.com
 ChangeSet@1.2136, 2005-01-25 20:56:30-08:00, olh@suse.de
 ChangeSet@1.1966.1.170, 2005-01-25 20:56:30-08:00, olh@suse.de
 ChangeSet@1.2135, 2005-01-25 20:56:12-08:00, rusty@rustcorp.com.au
 ChangeSet@1.1966.1.169, 2005-01-25 20:56:12-08:00, rusty@rustcorp.com.au
 ChangeSet@1.2134, 2005-01-25 20:55:59-08:00, adaplas@hotpop.com
 ChangeSet@1.1966.1.168, 2005-01-25 20:55:59-08:00, adaplas@hotpop.com
 ChangeSet@1.2133, 2005-01-25 20:55:42-08:00, adaplas@hotpop.com
 ChangeSet@1.1966.1.167, 2005-01-25 20:55:42-08:00, adaplas@hotpop.com
 ChangeSet@1.2132, 2005-01-25 20:55:28-08:00, adaplas@hotpop.com
 ChangeSet@1.1966.1.166, 2005-01-25 20:55:28-08:00, adaplas@hotpop.com
 ChangeSet@1.2131, 2005-01-25 20:55:14-08:00, adaplas@hotpop.com
 ChangeSet@1.1966.1.165, 2005-01-25 20:55:14-08:00, adaplas@hotpop.com
 ChangeSet@1.2130, 2005-01-25 20:54:57-08:00, adaplas@hotpop.com
 ChangeSet@1.1966.1.164, 2005-01-25 20:54:57-08:00, adaplas@hotpop.com
 ChangeSet@1.2129, 2005-01-25 20:54:43-08:00, adaplas@hotpop.com
 ChangeSet@1.1966.1.163, 2005-01-25 20:54:43-08:00, adaplas@hotpop.com
 ChangeSet@1.2128, 2005-01-25 20:54:24-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.162, 2005-01-25 20:54:24-08:00, kraxel@bytesex.org
 ChangeSet@1.2127, 2005-01-25 20:54:10-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.161, 2005-01-25 20:54:10-08:00, kraxel@bytesex.org
 ChangeSet@1.2126, 2005-01-25 20:53:56-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.160, 2005-01-25 20:53:56-08:00, kraxel@bytesex.org
 ChangeSet@1.2125, 2005-01-25 20:53:37-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.159, 2005-01-25 20:53:37-08:00, kraxel@bytesex.org
 ChangeSet@1.2124, 2005-01-25 20:53:22-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.158, 2005-01-25 20:53:22-08:00, kraxel@bytesex.org
 ChangeSet@1.2123, 2005-01-25 20:53:05-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.157, 2005-01-25 20:53:05-08:00, kraxel@bytesex.org
 ChangeSet@1.2122, 2005-01-25 20:52:50-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.156, 2005-01-25 20:52:50-08:00, kraxel@bytesex.org
 ChangeSet@1.2121, 2005-01-25 20:52:33-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.155, 2005-01-25 20:52:33-08:00, kraxel@bytesex.org
 ChangeSet@1.2120, 2005-01-25 20:52:17-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.154, 2005-01-25 20:52:17-08:00, kraxel@bytesex.org
 ChangeSet@1.2119, 2005-01-25 20:52:00-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.153, 2005-01-25 20:52:00-08:00, kraxel@bytesex.org
 ChangeSet@1.2118, 2005-01-25 20:51:45-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.152, 2005-01-25 20:51:45-08:00, kraxel@bytesex.org
 ChangeSet@1.2117, 2005-01-25 20:51:29-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.151, 2005-01-25 20:51:29-08:00, kraxel@bytesex.org
 ChangeSet@1.2116, 2005-01-25 20:51:15-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.150, 2005-01-25 20:51:15-08:00, kraxel@bytesex.org
 ChangeSet@1.2115, 2005-01-25 20:50:58-08:00, kraxel@bytesex.org
 ChangeSet@1.1966.1.149, 2005-01-25 20:50:58-08:00, kraxel@bytesex.org
 ChangeSet@1.2114, 2005-01-25 20:50:44-08:00, linux@dominikbrodowski.de
 ChangeSet@1.1966.1.148, 2005-01-25 20:50:44-08:00, linux@dominikbrodowski.de
 ChangeSet@1.2113, 2005-01-25 20:50:25-08:00, linux@dominikbrodowski.de
 ChangeSet@1.1966.1.147, 2005-01-25 20:50:25-08:00, linux@dominikbrodowski.de
 ChangeSet@1.2112, 2005-01-25 20:50:11-08:00, linux@dominikbrodowski.de
 ChangeSet@1.1966.1.146, 2005-01-25 20:50:11-08:00, linux@dominikbrodowski.de
 ChangeSet@1.2111, 2005-01-25 20:49:54-08:00, linux@dominikbrodowski.de
 ChangeSet@1.1966.1.145, 2005-01-25 20:49:54-08:00, linux@dominikbrodowski.de
 ChangeSet@1.2110, 2005-01-25 20:49:39-08:00, linux@dominikbrodowski.de
 ChangeSet@1.1966.1.144, 2005-01-25 20:49:39-08:00, linux@dominikbrodowski.de
 ChangeSet@1.2109, 2005-01-25 20:49:23-08:00, roland@topspin.com
 ChangeSet@1.1966.1.143, 2005-01-25 20:49:23-08:00, roland@topspin.com
 ChangeSet@1.2108, 2005-01-25 20:49:05-08:00, roland@topspin.com
 ChangeSet@1.1966.1.142, 2005-01-25 20:49:05-08:00, roland@topspin.com
 ChangeSet@1.2107, 2005-01-25 20:48:48-08:00, roland@topspin.com
 ChangeSet@1.1966.1.141, 2005-01-25 20:48:48-08:00, roland@topspin.com
 ChangeSet@1.2106, 2005-01-25 20:48:33-08:00, roland@topspin.com
 ChangeSet@1.1966.1.140, 2005-01-25 20:48:33-08:00, roland@topspin.com
 ChangeSet@1.2105, 2005-01-25 20:48:17-08:00, roland@topspin.com
 ChangeSet@1.1966.1.139, 2005-01-25 20:48:17-08:00, roland@topspin.com
 ChangeSet@1.2104, 2005-01-25 20:48:02-08:00, roland@topspin.com
 ChangeSet@1.1966.1.138, 2005-01-25 20:48:02-08:00, roland@topspin.com
 ChangeSet@1.2103, 2005-01-25 20:47:46-08:00, roland@topspin.com
 ChangeSet@1.1966.1.137, 2005-01-25 20:47:46-08:00, roland@topspin.com
 ChangeSet@1.2102, 2005-01-25 20:47:32-08:00, roland@topspin.com
 ChangeSet@1.1966.1.136, 2005-01-25 20:47:32-08:00, roland@topspin.com
 ChangeSet@1.2101, 2005-01-25 20:47:14-08:00, roland@topspin.com
 ChangeSet@1.1966.1.135, 2005-01-25 20:47:14-08:00, roland@topspin.com
 ChangeSet@1.2100, 2005-01-25 20:47:00-08:00, roland@topspin.com
 ChangeSet@1.1966.1.134, 2005-01-25 20:47:00-08:00, roland@topspin.com
 ChangeSet@1.2099, 2005-01-25 20:46:43-08:00, roland@topspin.com
 ChangeSet@1.1966.1.133, 2005-01-25 20:46:43-08:00, roland@topspin.com
 ChangeSet@1.2098, 2005-01-25 20:46:29-08:00, roland@topspin.com
 ChangeSet@1.1966.1.132, 2005-01-25 20:46:29-08:00, roland@topspin.com
 ChangeSet@1.2097, 2005-01-25 20:46:12-08:00, roland@topspin.com
 ChangeSet@1.1966.1.131, 2005-01-25 20:46:12-08:00, roland@topspin.com
 ChangeSet@1.2096, 2005-01-25 20:45:58-08:00, anton@samba.org
 ChangeSet@1.1966.1.130, 2005-01-25 20:45:58-08:00, anton@samba.org
 ChangeSet@1.2095, 2005-01-25 20:45:45-08:00, anton@samba.org
 ChangeSet@1.1966.1.129, 2005-01-25 20:45:45-08:00, anton@samba.org
 ChangeSet@1.2094, 2005-01-25 20:45:28-08:00, james4765@cwazy.co.uk
 ChangeSet@1.1966.1.128, 2005-01-25 20:45:28-08:00, james4765@cwazy.co.uk
 ChangeSet@1.2093, 2005-01-25 20:45:13-08:00, kevcorry@us.ibm.com
 ChangeSet@1.1966.1.127, 2005-01-25 20:45:13-08:00, kevcorry@us.ibm.com
 ChangeSet@1.2092, 2005-01-25 20:44:57-08:00, agk@redhat.com
 ChangeSet@1.1966.1.126, 2005-01-25 20:44:57-08:00, agk@redhat.com
 ChangeSet@1.2091, 2005-01-25 20:44:38-08:00, agk@redhat.com
 ChangeSet@1.1966.1.125, 2005-01-25 20:44:38-08:00, agk@redhat.com
 ChangeSet@1.2090, 2005-01-25 20:44:22-08:00, agk@redhat.com
 ChangeSet@1.1966.1.124, 2005-01-25 20:44:22-08:00, agk@redhat.com
 ChangeSet@1.2089, 2005-01-25 20:44:09-08:00, agk@redhat.com
 ChangeSet@1.1966.1.123, 2005-01-25 20:44:09-08:00, agk@redhat.com
 ChangeSet@1.2088, 2005-01-25 20:43:55-08:00, agruen@suse.de
 ChangeSet@1.1966.1.122, 2005-01-25 20:43:55-08:00, agruen@suse.de
 ChangeSet@1.2087, 2005-01-25 20:43:39-08:00, dsd@gentoo.org
 ChangeSet@1.1966.1.121, 2005-01-25 20:43:39-08:00, dsd@gentoo.org
 ChangeSet@1.2086, 2005-01-25 20:43:20-08:00, linville@tuxdriver.com
 ChangeSet@1.1966.1.120, 2005-01-25 20:43:20-08:00, linville@tuxdriver.com
 ChangeSet@1.2085, 2005-01-25 20:43:03-08:00, matthew@wil.cx
 ChangeSet@1.1966.1.119, 2005-01-25 20:43:03-08:00, matthew@wil.cx
 ChangeSet@1.2084, 2005-01-25 20:42:49-08:00, agruen@suse.de
 ChangeSet@1.1966.1.118, 2005-01-25 20:42:49-08:00, agruen@suse.de
 ChangeSet@1.2083, 2005-01-25 20:42:31-08:00, agruen@suse.de
 ChangeSet@1.1966.1.117, 2005-01-25 20:42:31-08:00, agruen@suse.de
 ChangeSet@1.2082, 2005-01-25 20:42:18-08:00, agruen@suse.de
 ChangeSet@1.1966.1.116, 2005-01-25 20:42:18-08:00, agruen@suse.de
 ChangeSet@1.2081, 2005-01-25 20:42:01-08:00, agruen@suse.de
 ChangeSet@1.1966.1.115, 2005-01-25 20:42:01-08:00, agruen@suse.de
 ChangeSet@1.2080, 2005-01-25 20:41:47-08:00, agruen@suse.de
 ChangeSet@1.1966.1.114, 2005-01-25 20:41:47-08:00, agruen@suse.de
 ChangeSet@1.2079, 2005-01-25 20:41:31-08:00, agruen@suse.de
 ChangeSet@1.1966.1.113, 2005-01-25 20:41:31-08:00, agruen@suse.de
 ChangeSet@1.2078, 2005-01-25 20:41:18-08:00, agruen@suse.de
 ChangeSet@1.1966.1.112, 2005-01-25 20:41:18-08:00, agruen@suse.de
 ChangeSet@1.2077, 2005-01-25 20:41:05-08:00, stone_wang@sohu.com
 ChangeSet@1.1966.1.111, 2005-01-25 20:41:05-08:00, stone_wang@sohu.com
 ChangeSet@1.2076, 2005-01-25 20:40:51-08:00, klassert@mathematik.tu-chemnitz.de
 ChangeSet@1.1966.1.110, 2005-01-25 20:40:51-08:00, klassert@mathematik.tu-chemnitz.de
 ChangeSet@1.2075, 2005-01-25 20:40:34-08:00, bgerst@didntduck.org
 ChangeSet@1.1966.1.109, 2005-01-25 20:40:34-08:00, bgerst@didntduck.org
 ChangeSet@1.2074, 2005-01-25 20:40:18-08:00, hch@lst.de
 ChangeSet@1.1966.1.108, 2005-01-25 20:40:18-08:00, hch@lst.de
 ChangeSet@1.2073, 2005-01-25 20:40:02-08:00, pavel@ucw.cz
 ChangeSet@1.1966.1.107, 2005-01-25 20:40:02-08:00, pavel@ucw.cz
 ChangeSet@1.2072, 2005-01-25 20:39:48-08:00, pavel@suse.cz
 ChangeSet@1.1966.1.106, 2005-01-25 20:39:48-08:00, pavel@suse.cz
 ChangeSet@1.2071, 2005-01-25 20:39:34-08:00, ysato@users.sourceforge.jp
 ChangeSet@1.1966.1.105, 2005-01-25 20:39:34-08:00, ysato@users.sourceforge.jp
 ChangeSet@1.2070, 2005-01-25 20:39:17-08:00, ysato@users.sourceforge.jp
 ChangeSet@1.1966.1.104, 2005-01-25 20:39:17-08:00, ysato@users.sourceforge.jp
 ChangeSet@1.2069, 2005-01-25 20:39:03-08:00, ak@suse.de
 ChangeSet@1.1966.1.103, 2005-01-25 20:39:03-08:00, ak@suse.de
 ChangeSet@1.2068, 2005-01-25 20:38:49-08:00, ak@suse.de
 ChangeSet@1.1966.1.102, 2005-01-25 20:38:49-08:00, ak@suse.de
 ChangeSet@1.2067, 2005-01-25 20:38:31-08:00, ak@suse.de
 ChangeSet@1.1966.1.101, 2005-01-25 20:38:31-08:00, ak@suse.de
 ChangeSet@1.2066, 2005-01-25 20:38:18-08:00, ak@suse.de
 ChangeSet@1.1966.1.100, 2005-01-25 20:38:18-08:00, ak@suse.de
 ChangeSet@1.2065, 2005-01-25 20:38:01-08:00, ak@suse.de
 ChangeSet@1.1966.1.99, 2005-01-25 20:38:01-08:00, ak@suse.de
 ChangeSet@1.2064, 2005-01-25 20:37:47-08:00, ak@suse.de
 ChangeSet@1.1966.1.98, 2005-01-25 20:37:47-08:00, ak@suse.de
 ChangeSet@1.2063, 2005-01-25 20:37:31-08:00, rusty@rustcorp.com.au
 ChangeSet@1.1966.1.97, 2005-01-25 20:37:31-08:00, rusty@rustcorp.com.au
 ChangeSet@1.2062, 2005-01-25 20:37:16-08:00, benh@kernel.crashing.org
 ChangeSet@1.1966.1.96, 2005-01-25 20:37:16-08:00, benh@kernel.crashing.org
 ChangeSet@1.2061, 2005-01-25 20:37:00-08:00, anton@samba.org
 ChangeSet@1.1966.1.95, 2005-01-25 20:37:00-08:00, anton@samba.org
 ChangeSet@1.2060, 2005-01-25 20:36:45-08:00, anton@samba.org
 ChangeSet@1.1966.1.94, 2005-01-25 20:36:45-08:00, anton@samba.org
 ChangeSet@1.2059, 2005-01-25 20:36:29-08:00, anton@samba.org
 ChangeSet@1.1966.1.93, 2005-01-25 20:36:29-08:00, anton@samba.org
 ChangeSet@1.2058, 2005-01-25 20:36:14-08:00, anton@samba.org
 ChangeSet@1.1966.1.92, 2005-01-25 20:36:14-08:00, anton@samba.org
 ChangeSet@1.2057, 2005-01-25 20:35:57-08:00, paulus@samba.org
 ChangeSet@1.1966.1.91, 2005-01-25 20:35:57-08:00, paulus@samba.org
 ChangeSet@1.2056, 2005-01-25 20:35:44-08:00, paulus@samba.org
 ChangeSet@1.1966.1.90, 2005-01-25 20:35:44-08:00, paulus@samba.org
 ChangeSet@1.2055, 2005-01-25 20:35:29-08:00, paulus@samba.org
 ChangeSet@1.1966.1.89, 2005-01-25 20:35:29-08:00, paulus@samba.org
 ChangeSet@1.2054, 2005-01-25 20:35:13-08:00, paulus@samba.org
 ChangeSet@1.1966.1.88, 2005-01-25 20:35:13-08:00, paulus@samba.org
 ChangeSet@1.2053, 2005-01-25 20:34:59-08:00, paulus@samba.org
 ChangeSet@1.1966.1.87, 2005-01-25 20:34:59-08:00, paulus@samba.org
 ChangeSet@1.2052, 2005-01-25 20:34:42-08:00, paulus@samba.org
 ChangeSet@1.1966.1.86, 2005-01-25 20:34:42-08:00, paulus@samba.org
 ChangeSet@1.2051, 2005-01-25 20:34:27-08:00, paulus@samba.org
 ChangeSet@1.1966.1.85, 2005-01-25 20:34:27-08:00, paulus@samba.org
 ChangeSet@1.2050, 2005-01-25 20:34:11-08:00, paulus@samba.org
 ChangeSet@1.1966.1.84, 2005-01-25 20:34:11-08:00, paulus@samba.org
 ChangeSet@1.2049, 2005-01-25 20:33:57-08:00, paulus@samba.org
 ChangeSet@1.1966.1.83, 2005-01-25 20:33:57-08:00, paulus@samba.org
 ChangeSet@1.2048, 2005-01-25 20:33:40-08:00, paulus@samba.org
 ChangeSet@1.1966.1.82, 2005-01-25 20:33:40-08:00, paulus@samba.org
 ChangeSet@1.2047, 2005-01-25 20:33:26-08:00, anton@samba.org
 ChangeSet@1.1966.1.81, 2005-01-25 20:33:26-08:00, anton@samba.org
 ChangeSet@1.2046, 2005-01-25 20:33:08-08:00, benh@kernel.crashing.org
 ChangeSet@1.1966.1.80, 2005-01-25 20:33:08-08:00, benh@kernel.crashing.org
 ChangeSet@1.2045, 2005-01-25 20:32:53-08:00, galak@freescale.com
 ChangeSet@1.1966.1.79, 2005-01-25 20:32:53-08:00, galak@freescale.com
 ChangeSet@1.2044, 2005-01-25 20:32:37-08:00, rvinson@mvista.com
 ChangeSet@1.1966.1.78, 2005-01-25 20:32:37-08:00, rvinson@mvista.com
 ChangeSet@1.2043, 2005-01-25 20:32:23-08:00, galak@freescale.com
 ChangeSet@1.1966.1.77, 2005-01-25 20:32:23-08:00, galak@freescale.com
 ChangeSet@1.2042, 2005-01-25 20:32:06-08:00, galak@freescale.com
 ChangeSet@1.1966.1.76, 2005-01-25 20:32:06-08:00, galak@freescale.com
 ChangeSet@1.2041, 2005-01-25 20:31:51-08:00, andrea@suse.de
 ChangeSet@1.1966.1.75, 2005-01-25 20:31:51-08:00, andrea@suse.de
 ChangeSet@1.2040, 2005-01-25 20:31:34-08:00, andrea@suse.de
 ChangeSet@1.1966.1.74, 2005-01-25 20:31:34-08:00, andrea@suse.de
 ChangeSet@1.2039, 2005-01-25 20:31:20-08:00, chrisw@osdl.org
 ChangeSet@1.1966.1.73, 2005-01-25 20:31:20-08:00, chrisw@osdl.org
 ChangeSet@1.2038, 2005-01-25 20:31:01-08:00, amgta@yacht.ocn.ne.jp
 ChangeSet@1.1966.1.72, 2005-01-25 20:31:01-08:00, amgta@yacht.ocn.ne.jp
 ChangeSet@1.2037, 2005-01-25 20:30:47-08:00, yuasa@hh.iij4u.or.jp
 ChangeSet@1.1966.1.71, 2005-01-25 20:30:47-08:00, yuasa@hh.iij4u.or.jp
 ChangeSet@1.2036, 2005-01-25 20:30:29-08:00, js@linuxtv.org
 ChangeSet@1.1966.1.70, 2005-01-25 20:30:29-08:00, js@linuxtv.org
 ChangeSet@1.2035, 2005-01-25 20:30:15-08:00, akpm@osdl.org
 ChangeSet@1.1966.1.69, 2005-01-25 20:30:15-08:00, akpm@osdl.org
 ChangeSet@1.2034, 2005-01-25 20:29:47-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.1.68, 2005-01-25 20:29:47-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.2033, 2005-01-25 15:00:33-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.1.67, 2005-01-25 15:00:33-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.2032, 2005-01-25 12:49:07-08:00, greg@kroah.com
 ChangeSet@1.1966.85.3, 2005-01-25 22:51:34+00:00, nico@org.rmk.(none)
   [ARM PATCH] 2435/1: platform data for audio on Mainstone
   
   Patch from Nicolas Pitre
   
   This enables audio on Mainstone with the ALSA PXA2xx AC97 driver.
   
   Signed-off-by: Nicolas Pitre
   Signed-off-by: Russell King
 
 ChangeSet@1.1966.85.2, 2005-01-25 22:43:59+00:00, ben-linux@org.rmk.(none)
   [ARM PATCH] 2433/1: debug-macro.S split - fix header filename
   
   Patch from Ben Dooks
   
   All the headers refer to debug-armv.S, when they where split from
   debug.S
   
   Signed-off-by: Ben Dooks
   Signed-off-by: Russell King
 
 ChangeSet@1.1966.85.1, 2005-01-25 22:36:39+00:00, ben-linux@org.rmk.(none)
   [ARM PATCH] 2432/1: include/asm-arm/bitops.h - missing const from find
   
   Patch from Ben Dooks
   
   the calls _find_first_zero_bit_be() and _find_next_zero_bit_be()
   are missing const from the pointer argument passed in, causing
   several build warnings when building an ARM kernel in big-endian
   mode.
   
   Signed-off-by: Ben Dooks
   Signed-off-by: Russell King
 
 ChangeSet@1.1966.1.66, 2005-01-25 12:49:07-08:00, greg@kroah.com
 ChangeSet@1.2031, 2005-01-25 08:03:42-08:00, geert@linux-m68k.org
 ChangeSet@1.1966.1.65, 2005-01-25 08:03:42-08:00, geert@linux-m68k.org
 ChangeSet@1.2030, 2005-01-25 08:01:03-08:00, ecashin@coraid.com
 ChangeSet@1.1966.1.64, 2005-01-25 08:01:03-08:00, ecashin@coraid.com
 ChangeSet@1.1983.6.1, 2005-01-25 10:10:51+00:00, aia21@cantab.net
 ChangeSet@1.1966.83.1, 2005-01-25 10:10:51+00:00, aia21@cantab.net
 ChangeSet@1.2020.2.12, 2005-01-25 04:31:49+01:00, kernel@linuxace.com
 ChangeSet@1.1966.82.12, 2005-01-25 04:31:49+01:00, kernel@linuxace.com
 ChangeSet@1.2020.2.11, 2005-01-25 04:12:44+01:00, gandalf@wlug.westbo.se
 ChangeSet@1.1966.82.11, 2005-01-25 04:12:44+01:00, gandalf@wlug.westbo.se
 ChangeSet@1.2020.2.10, 2005-01-25 04:10:59+01:00, gandalf@wlug.westbo.se
 ChangeSet@1.1966.82.10, 2005-01-25 04:10:59+01:00, gandalf@wlug.westbo.se
 ChangeSet@1.2020.2.9, 2005-01-25 03:56:48+01:00, gandalf@wlug.westbo.se
 ChangeSet@1.1966.82.9, 2005-01-25 03:56:48+01:00, gandalf@wlug.westbo.se
 ChangeSet@1.2020.2.8, 2005-01-24 17:41:45-08:00, davem@nuts.davemloft.net
 ChangeSet@1.1966.82.8, 2005-01-24 17:41:45-08:00, davem@nuts.davemloft.net
 ChangeSet@1.2020.2.7, 2005-01-24 17:39:51-08:00, mchan@broadcom.com
 ChangeSet@1.1966.82.7, 2005-01-24 17:39:51-08:00, mchan@broadcom.com
 ChangeSet@1.2020.2.6, 2005-01-24 16:40:33-08:00, herbert@gondor.apana.org.au
 ChangeSet@1.1966.82.6, 2005-01-24 16:40:33-08:00, herbert@gondor.apana.org.au
 ChangeSet@1.2027, 2005-01-24 15:57:20-08:00, arjan@infradead.org
 ChangeSet@1.1966.1.61, 2005-01-24 15:57:20-08:00, arjan@infradead.org
 ChangeSet@1.2026, 2005-01-24 14:37:53-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.1.60, 2005-01-24 14:37:53-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.2025, 2005-01-24 14:36:59-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.1.59, 2005-01-24 14:36:59-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.2020.2.5, 2005-01-23 11:09:59-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.82.5, 2005-01-23 11:09:59-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.2020.1.5, 2005-01-23 20:01:39+01:00, marcel@holtmann.org
 ChangeSet@1.1966.81.5, 2005-01-23 20:01:39+01:00, marcel@holtmann.org
 ChangeSet@1.2020.1.4, 2005-01-23 19:55:30+01:00, marcel@holtmann.org
 ChangeSet@1.1966.81.4, 2005-01-23 19:55:30+01:00, marcel@holtmann.org
 ChangeSet@1.2020.2.4, 2005-01-23 10:29:42-08:00, kaos@ocs.com.au
 ChangeSet@1.1966.82.4, 2005-01-23 10:29:42-08:00, kaos@ocs.com.au
 ChangeSet@1.2020.2.3, 2005-01-23 10:29:27-08:00, ak@suse.de
 ChangeSet@1.1966.82.3, 2005-01-23 10:29:27-08:00, ak@suse.de
 ChangeSet@1.2020.2.2, 2005-01-23 10:29:11-08:00, ak@suse.de
 ChangeSet@1.1966.82.2, 2005-01-23 10:29:11-08:00, ak@suse.de
 ChangeSet@1.2020.2.1, 2005-01-23 10:28:57-08:00, ak@suse.de
 ChangeSet@1.1966.82.1, 2005-01-23 10:28:57-08:00, ak@suse.de
 ChangeSet@1.2020.1.3, 2005-01-23 18:50:26+01:00, marcel@holtmann.org
 ChangeSet@1.1966.81.3, 2005-01-23 18:50:26+01:00, marcel@holtmann.org
 ChangeSet@1.2023, 2005-01-23 17:47:02+00:00, ben-linux@org.rmk.(none)
 ChangeSet@1.1966.1.57, 2005-01-23 17:47:02+00:00, ben-linux@org.rmk.(none)
 ChangeSet@1.2020.1.2, 2005-01-23 18:40:28+01:00, marcel@holtmann.org
 ChangeSet@1.1966.81.2, 2005-01-23 18:40:28+01:00, marcel@holtmann.org
 ChangeSet@1.2020.1.1, 2005-01-23 18:36:35+01:00, marcel@holtmann.org
 ChangeSet@1.1966.81.1, 2005-01-23 18:36:35+01:00, marcel@holtmann.org
 ChangeSet@1.2022, 2005-01-23 13:45:48+00:00, rmk@flint.arm.linux.org.uk
 ChangeSet@1.1966.1.56, 2005-01-23 13:45:48+00:00, rmk@flint.arm.linux.org.uk
 ChangeSet@1.2021, 2005-01-23 13:33:19+00:00, rmk@flint.arm.linux.org.uk
 ChangeSet@1.1966.1.55, 2005-01-23 13:33:19+00:00, rmk@flint.arm.linux.org.uk
 ChangeSet@1.1983.2.145, 2005-01-22 19:46:13-08:00, davem@nuts.davemloft.net
 ChangeSet@1.1966.76.145, 2005-01-22 19:46:13-08:00, davem@nuts.davemloft.net
 ChangeSet@1.1983.1.19, 2005-01-22 19:42:06-08:00, davem@nuts.davemloft.net
 ChangeSet@1.1966.74.19, 2005-01-22 19:42:06-08:00, davem@nuts.davemloft.net
 ChangeSet@1.2017, 2005-01-22 16:01:24-08:00, tony.luck@intel.com
 ChangeSet@1.1966.1.51, 2005-01-22 16:01:24-08:00, tony.luck@intel.com
 ChangeSet@1.2016, 2005-01-22 15:56:39-08:00, tony.luck@intel.com
 ChangeSet@1.1966.1.50, 2005-01-22 15:56:39-08:00, tony.luck@intel.com
 ChangeSet@1.2015, 2005-01-22 15:00:59-08:00, jbarnes@sgi.com
 ChangeSet@1.1966.1.49, 2005-01-22 15:00:59-08:00, jbarnes@sgi.com
 ChangeSet@1.2014, 2005-01-22 14:58:21-08:00, jbarnes@sgi.com
 ChangeSet@1.1966.1.48, 2005-01-22 14:58:21-08:00, jbarnes@sgi.com
 ChangeSet@1.2013, 2005-01-22 14:45:15-08:00, jbarnes@sgi.com
 ChangeSet@1.1966.1.47, 2005-01-22 14:45:15-08:00, jbarnes@sgi.com
 ChangeSet@1.2012, 2005-01-22 14:39:46-08:00, jes@trained-monkey.org
 ChangeSet@1.1966.1.46, 2005-01-22 14:39:46-08:00, jes@trained-monkey.org
 ChangeSet@1.2011, 2005-01-22 14:35:36-08:00, eranian@hpl.hp.com
 ChangeSet@1.1966.1.45, 2005-01-22 14:35:36-08:00, eranian@hpl.hp.com
 ChangeSet@1.2010, 2005-01-22 14:21:34-08:00, tony.luck@intel.com
 ChangeSet@1.1966.1.44, 2005-01-22 14:21:34-08:00, tony.luck@intel.com
 ChangeSet@1.2009, 2005-01-22 13:50:44-08:00, tony.luck@intel.com
 ChangeSet@1.1966.1.43, 2005-01-22 13:50:44-08:00, tony.luck@intel.com
 ChangeSet@1.2008, 2005-01-22 13:35:36-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.42, 2005-01-22 13:35:36-08:00, davidm@hpl.hp.com
 ChangeSet@1.2007, 2005-01-22 13:33:33-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.41, 2005-01-22 13:33:33-08:00, davidm@hpl.hp.com
 ChangeSet@1.2006, 2005-01-22 13:31:05-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.40, 2005-01-22 13:31:05-08:00, davidm@hpl.hp.com
 ChangeSet@1.2005, 2005-01-22 13:29:13-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.39, 2005-01-22 13:29:13-08:00, davidm@hpl.hp.com
 ChangeSet@1.2004, 2005-01-22 13:27:29-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.38, 2005-01-22 13:27:29-08:00, davidm@hpl.hp.com
 ChangeSet@1.2003, 2005-01-22 13:25:31-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.37, 2005-01-22 13:25:31-08:00, davidm@hpl.hp.com
 ChangeSet@1.2002, 2005-01-22 13:22:55-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.36, 2005-01-22 13:22:55-08:00, davidm@hpl.hp.com
 ChangeSet@1.2001, 2005-01-22 13:16:18-08:00, nacc@us.ibm.com
 ChangeSet@1.1966.1.35, 2005-01-22 13:16:18-08:00, nacc@us.ibm.com
 ChangeSet@1.2000, 2005-01-22 13:09:59-08:00, rja@sgi.com
 ChangeSet@1.1966.1.34, 2005-01-22 13:09:59-08:00, rja@sgi.com
 ChangeSet@1.1999, 2005-01-22 13:03:24-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.33, 2005-01-22 13:03:24-08:00, davidm@hpl.hp.com
 ChangeSet@1.1998, 2005-01-22 12:59:23-08:00, kaos@sgi.com
 ChangeSet@1.1966.1.32, 2005-01-22 12:59:23-08:00, kaos@sgi.com
 ChangeSet@1.1997, 2005-01-22 12:51:34-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.31, 2005-01-22 12:51:34-08:00, davidm@hpl.hp.com
 ChangeSet@1.1996, 2005-01-22 12:46:27-08:00, davidm@hpl.hp.com
 ChangeSet@1.1966.1.30, 2005-01-22 12:46:27-08:00, davidm@hpl.hp.com
 ChangeSet@1.1995, 2005-01-22 12:18:41-08:00, tony.luck@intel.com
 ChangeSet@1.1966.1.29, 2005-01-22 12:18:41-08:00, tony.luck@intel.com
 ChangeSet@1.1994, 2005-01-22 12:08:41-08:00, markgw@sgi.com
 ChangeSet@1.1966.1.28, 2005-01-22 12:08:41-08:00, markgw@sgi.com
 ChangeSet@1.1993, 2005-01-22 12:03:07-08:00, steiner@sgi.com
 ChangeSet@1.1966.1.27, 2005-01-22 12:03:07-08:00, steiner@sgi.com
 ChangeSet@1.1992, 2005-01-22 11:57:39-08:00, steiner@sgi.com
 ChangeSet@1.1966.1.26, 2005-01-22 11:57:39-08:00, steiner@sgi.com
 ChangeSet@1.1991, 2005-01-22 11:50:47-08:00, bob.picco@hp.com
 ChangeSet@1.1966.1.25, 2005-01-22 11:50:47-08:00, bob.picco@hp.com
 ChangeSet@1.1983.4.16, 2005-01-22 11:37:01-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.16, 2005-01-22 11:37:01-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.15, 2005-01-22 11:36:47-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.15, 2005-01-22 11:36:47-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.14, 2005-01-22 11:36:33-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.14, 2005-01-22 11:36:33-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.13, 2005-01-22 11:36:18-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.13, 2005-01-22 11:36:18-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.12, 2005-01-22 11:36:05-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.12, 2005-01-22 11:36:05-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.11, 2005-01-22 11:35:52-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.11, 2005-01-22 11:35:52-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.10, 2005-01-22 11:35:38-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.10, 2005-01-22 11:35:38-08:00, js@linuxtv.org
 ChangeSet@1.1983.4.9, 2005-01-22 11:35:24-08:00, js@linuxtv.org
 ChangeSet@1.1966.79.9, 2005-01-22 11:35:24-08:00, js@linuxtv.org
 ChangeSet@1.1990, 2005-01-22 11:32:44-08:00, tony.luck@intel.com
 ChangeSet@1.1966.1.24, 2005-01-22 11:32:44-08:00, tony.luck@intel.com
 ChangeSet@1.1983.4.8, 2005-01-22 11:02:00-08:00, roland@redhat.com
 ChangeSet@1.1966.79.8, 2005-01-22 11:02:00-08:00, roland@redhat.com
 ChangeSet@1.1983.4.7, 2005-01-22 10:31:39-08:00, nanhai.zou@intel.com
 ChangeSet@1.1966.79.7, 2005-01-22 10:31:39-08:00, nanhai.zou@intel.com
 ChangeSet@1.1983.2.144, 2005-01-21 20:04:07-08:00, davem@nuts.davemloft.net
 ChangeSet@1.1974, 2005-01-22 11:19:29+01:00, vojtech@silver.ucw.cz
   Merge silver.ucw.cz:/home/vojtech/bk/linus
   into silver.ucw.cz:/home/vojtech/bk/input
 
 ChangeSet@1.1966.76.144, 2005-01-21 20:04:07-08:00, davem@nuts.davemloft.net
 ChangeSet@1.1983.4.6, 2005-01-21 17:45:41-08:00, torvalds@ppc970.osdl.org
 ChangeSet@1.1966.79.6, 2005-01-21 17:45:41-08:00, torvalds@ppc970.osdl.org
diff -purN linux-2.6.11-rc2-bk5/Makefile linux-2.6.11-rc2-bk6/Makefile
--- linux-2.6.11-rc2-bk5/Makefile	2005-01-28 15:50:05.478927500 +0100
+++ linux-2.6.11-rc2-bk6/Makefile	2005-01-28 15:50:13.168929473 +0100
@@ -1,7 +1,7 @@
 VERSION = 2
 PATCHLEVEL = 6
 SUBLEVEL = 11
-EXTRAVERSION = -rc2-bk5
+EXTRAVERSION = -rc2-bk6
 NAME=Woozy Numbat
 
 # *DOCUMENTATION*
diff -purN linux-2.6.11-rc2-bk5/arch/arm/kernel/entry-armv.S linux-2.6.11-rc2-bk6/arch/arm/kernel/entry-armv.S
--- linux-2.6.11-rc2-bk5/arch/arm/kernel/entry-armv.S	2005-01-22 02:47:32.000000000 +0100
+++ linux-2.6.11-rc2-bk6/arch/arm/kernel/entry-armv.S	2005-01-28 15:50:13.182927299 +0100
@@ -429,186 +429,131 @@ ENTRY(__switch_to)
 
 		__INIT
 /*
- * Vector stubs.  NOTE that we only align 'vector_IRQ' to a cache line boundary,
- * and we rely on each stub being exactly 48 (1.5 cache lines) in size.  This
- * means that we only ever load two cache lines for this code, or one if we're
- * lucky.  We also copy this code to 0x200 so that we can use branches in the
+ * Vector stubs.
+ *
+ * This code is copied to 0x200 or 0xffff0200 so we can use branches in the
  * vectors, rather than ldr's.
- */
-		.align	5
+ *
+ * Common stub entry macro:
+ *   Enter in IRQ mode, spsr = SVC/USR CPSR, lr = SVC/USR PC
+ */
+	.macro	vector_stub, name, sym, correction=0
+	.align	5
+
+vector_\name:
+	ldr	r13, .LCs\sym
+	.if \correction
+	sub	lr, lr, #\correction
+	.endif
+	str	lr, [r13]			@ save lr_IRQ
+	mrs	lr, spsr
+	str	lr, [r13, #4]			@ save spsr_IRQ
+	@
+	@ now branch to the relevant MODE handling routine
+	@
+	mrs	r13, cpsr
+	bic	r13, r13, #MODE_MASK
+	orr	r13, r13, #MODE_SVC
+	msr	spsr_cxsf, r13			@ switch to SVC_32 mode
+
+	and	lr, lr, #15
+	ldr	lr, [pc, lr, lsl #2]
+	movs	pc, lr				@ Changes mode and branches
+	.endm
+
 __stubs_start:
 /*
  * Interrupt dispatcher
- * Enter in IRQ mode, spsr = SVC/USR CPSR, lr = SVC/USR PC
  */
-vector_IRQ:	@
-		@ save mode specific registers
-		@
-		ldr	r13, .LCsirq
-		sub	lr, lr, #4
-		str	lr, [r13]			@ save lr_IRQ
-		mrs	lr, spsr
-		str	lr, [r13, #4]			@ save spsr_IRQ
-		@
-		@ now branch to the relevant MODE handling routine
-		@
-		mrs	r13, cpsr
-		bic	r13, r13, #MODE_MASK
-		orr	r13, r13, #MODE_SVC
-		msr	spsr_cxsf, r13			@ switch to SVC_32 mode
-
-		and	lr, lr, #15
-		ldr	lr, [pc, lr, lsl #2]
-		movs	pc, lr				@ Changes mode and branches
-
-.LCtab_irq:	.word	__irq_usr			@  0  (USR_26 / USR_32)
-		.word	__irq_invalid			@  1  (FIQ_26 / FIQ_32)
-		.word	__irq_invalid			@  2  (IRQ_26 / IRQ_32)
-		.word	__irq_svc			@  3  (SVC_26 / SVC_32)
-		.word	__irq_invalid			@  4
-		.word	__irq_invalid			@  5
-		.word	__irq_invalid			@  6
-		.word	__irq_invalid			@  7
-		.word	__irq_invalid			@  8
-		.word	__irq_invalid			@  9
-		.word	__irq_invalid			@  a
-		.word	__irq_invalid			@  b
-		.word	__irq_invalid			@  c
-		.word	__irq_invalid			@  d
-		.word	__irq_invalid			@  e
-		.word	__irq_invalid			@  f
+	vector_stub	irq, irq, 4
 
-		.align	5
+	.long	__irq_usr			@  0  (USR_26 / USR_32)
+	.long	__irq_invalid			@  1  (FIQ_26 / FIQ_32)
+	.long	__irq_invalid			@  2  (IRQ_26 / IRQ_32)
+	.long	__irq_svc			@  3  (SVC_26 / SVC_32)
+	.long	__irq_invalid			@  4
+	.long	__irq_invalid			@  5
+	.long	__irq_invalid			@  6
+	.long	__irq_invalid			@  7
+	.long	__irq_invalid			@  8
+	.long	__irq_invalid			@  9
+	.long	__irq_invalid			@  a
+	.long	__irq_invalid			@  b
+	.long	__irq_invalid			@  c
+	.long	__irq_invalid			@  d
+	.long	__irq_invalid			@  e
+	.long	__irq_invalid			@  f
 
 /*
- * Data abort dispatcher - dispatches it to the correct handler for the processor mode
+ * Data abort dispatcher
  * Enter in ABT mode, spsr = USR CPSR, lr = USR PC
  */
-vector_data:	@
-		@ save mode specific registers
-		@
-		ldr	r13, .LCsabt
-		sub	lr, lr, #8
-		str	lr, [r13]
-		mrs	lr, spsr
-		str	lr, [r13, #4]
-		@
-		@ now branch to the relevant MODE handling routine
-		@
-		mrs	r13, cpsr
-		bic	r13, r13, #MODE_MASK
-		orr	r13, r13, #MODE_SVC
-		msr	spsr_cxsf, r13			@ switch to SVC_32 mode
-
-		and	lr, lr, #15
-		ldr	lr, [pc, lr, lsl #2]
-		movs	pc, lr				@ Changes mode and branches
-
-.LCtab_dabt:	.word	__dabt_usr			@  0  (USR_26 / USR_32)
-		.word	__dabt_invalid			@  1  (FIQ_26 / FIQ_32)
-		.word	__dabt_invalid			@  2  (IRQ_26 / IRQ_32)
-		.word	__dabt_svc			@  3  (SVC_26 / SVC_32)
-		.word	__dabt_invalid			@  4
-		.word	__dabt_invalid			@  5
-		.word	__dabt_invalid			@  6
-		.word	__dabt_invalid			@  7
-		.word	__dabt_invalid			@  8
-		.word	__dabt_invalid			@  9
-		.word	__dabt_invalid			@  a
-		.word	__dabt_invalid			@  b
-		.word	__dabt_invalid			@  c
-		.word	__dabt_invalid			@  d
-		.word	__dabt_invalid			@  e
-		.word	__dabt_invalid			@  f
+	vector_stub	dabt, abt, 8
 
-		.align	5
+	.long	__dabt_usr			@  0  (USR_26 / USR_32)
+	.long	__dabt_invalid			@  1  (FIQ_26 / FIQ_32)
+	.long	__dabt_invalid			@  2  (IRQ_26 / IRQ_32)
+	.long	__dabt_svc			@  3  (SVC_26 / SVC_32)
+	.long	__dabt_invalid			@  4
+	.long	__dabt_invalid			@  5
+	.long	__dabt_invalid			@  6
+	.long	__dabt_invalid			@  7
+	.long	__dabt_invalid			@  8
+	.long	__dabt_invalid			@  9
+	.long	__dabt_invalid			@  a
+	.long	__dabt_invalid			@  b
+	.long	__dabt_invalid			@  c
+	.long	__dabt_invalid			@  d
+	.long	__dabt_invalid			@  e
+	.long	__dabt_invalid			@  f
 
 /*
- * Prefetch abort dispatcher - dispatches it to the correct handler for the processor mode
+ * Prefetch abort dispatcher
  * Enter in ABT mode, spsr = USR CPSR, lr = USR PC
  */
-vector_prefetch:
-		@
-		@ save mode specific registers
-		@
-		ldr	r13, .LCsabt
-		sub	lr, lr, #4
-		str	lr, [r13]			@ save lr_ABT
-		mrs	lr, spsr
-		str	lr, [r13, #4]			@ save spsr_ABT
-		@
-		@ now branch to the relevant MODE handling routine
-		@
-		mrs	r13, cpsr
-		bic	r13, r13, #MODE_MASK
-		orr	r13, r13, #MODE_SVC
-		msr	spsr_cxsf, r13			@ switch to SVC_32 mode
-
-		ands	lr, lr, #15
-		ldr	lr, [pc, lr, lsl #2]
-		movs	pc, lr
-
-.LCtab_pabt:	.word	__pabt_usr			@  0 (USR_26 / USR_32)
-		.word	__pabt_invalid			@  1 (FIQ_26 / FIQ_32)
-		.word	__pabt_invalid			@  2 (IRQ_26 / IRQ_32)
-		.word	__pabt_svc			@  3 (SVC_26 / SVC_32)
-		.word	__pabt_invalid			@  4
-		.word	__pabt_invalid			@  5
-		.word	__pabt_invalid			@  6
-		.word	__pabt_invalid			@  7
-		.word	__pabt_invalid			@  8
-		.word	__pabt_invalid			@  9
-		.word	__pabt_invalid			@  a
-		.word	__pabt_invalid			@  b
-		.word	__pabt_invalid			@  c
-		.word	__pabt_invalid			@  d
-		.word	__pabt_invalid			@  e
-		.word	__pabt_invalid			@  f
+	vector_stub	pabt, abt, 4
 
-		.align	5
+	.long	__pabt_usr			@  0 (USR_26 / USR_32)
+	.long	__pabt_invalid			@  1 (FIQ_26 / FIQ_32)
+	.long	__pabt_invalid			@  2 (IRQ_26 / IRQ_32)
+	.long	__pabt_svc			@  3 (SVC_26 / SVC_32)
+	.long	__pabt_invalid			@  4
+	.long	__pabt_invalid			@  5
+	.long	__pabt_invalid			@  6
+	.long	__pabt_invalid			@  7
+	.long	__pabt_invalid			@  8
+	.long	__pabt_invalid			@  9
+	.long	__pabt_invalid			@  a
+	.long	__pabt_invalid			@  b
+	.long	__pabt_invalid			@  c
+	.long	__pabt_invalid			@  d
+	.long	__pabt_invalid			@  e
+	.long	__pabt_invalid			@  f
 
 /*
- * Undef instr entry dispatcher - dispatches it to the correct handler for the processor mode
+ * Undef instr entry dispatcher
  * Enter in UND mode, spsr = SVC/USR CPSR, lr = SVC/USR PC
  */
-vector_undefinstr:
-		@
-		@ save mode specific registers
-		@
-		ldr	r13, .LCsund
-		str	lr, [r13]			@ save lr_UND
-		mrs	lr, spsr
-		str	lr, [r13, #4]			@ save spsr_UND
-		@
-		@ now branch to the relevant MODE handling routine
-		@
-		mrs	r13, cpsr
-		bic	r13, r13, #MODE_MASK
-		orr	r13, r13, #MODE_SVC
-		msr	spsr_cxsf, r13			@ switch to SVC_32 mode
-
-		and	lr, lr, #15
-		ldr	lr, [pc, lr, lsl #2]
-		movs	pc, lr				@ Changes mode and branches
-
-.LCtab_und:	.word	__und_usr			@  0 (USR_26 / USR_32)
-		.word	__und_invalid			@  1 (FIQ_26 / FIQ_32)
-		.word	__und_invalid			@  2 (IRQ_26 / IRQ_32)
-		.word	__und_svc			@  3 (SVC_26 / SVC_32)
-		.word	__und_invalid			@  4
-		.word	__und_invalid			@  5
-		.word	__und_invalid			@  6
-		.word	__und_invalid			@  7
-		.word	__und_invalid			@  8
-		.word	__und_invalid			@  9
-		.word	__und_invalid			@  a
-		.word	__und_invalid			@  b
-		.word	__und_invalid			@  c
-		.word	__und_invalid			@  d
-		.word	__und_invalid			@  e
-		.word	__und_invalid			@  f
+	vector_stub	und, und
 
-		.align	5
+	.long	__und_usr			@  0 (USR_26 / USR_32)
+	.long	__und_invalid			@  1 (FIQ_26 / FIQ_32)
+	.long	__und_invalid			@  2 (IRQ_26 / IRQ_32)
+	.long	__und_svc			@  3 (SVC_26 / SVC_32)
+	.long	__und_invalid			@  4
+	.long	__und_invalid			@  5
+	.long	__und_invalid			@  6
+	.long	__und_invalid			@  7
+	.long	__und_invalid			@  8
+	.long	__und_invalid			@  9
+	.long	__und_invalid			@  a
+	.long	__und_invalid			@  b
+	.long	__und_invalid			@  c
+	.long	__und_invalid			@  d
+	.long	__und_invalid			@  e
+	.long	__und_invalid			@  f
+
+	.align	5
 
 /*=============================================================================
  * Undefined FIQs
@@ -620,7 +565,7 @@ vector_undefinstr:
  * other mode than FIQ...  Ok you can switch to another mode, but you can't
  * get out of that mode without clobbering one register.
  */
-vector_FIQ:	disable_fiq
+vector_fiq:	disable_fiq
 		subs	pc, lr, #4
 
 /*=============================================================================
@@ -650,13 +595,13 @@ __stubs_end:
 		.equ	__real_stubs_start, .LCvectors + 0x200
 
 .LCvectors:	swi	SYS_ERROR0
-		b	__real_stubs_start + (vector_undefinstr - __stubs_start)
+		b	__real_stubs_start + (vector_und - __stubs_start)
 		ldr	pc, __real_stubs_start + (.LCvswi - __stubs_start)
-		b	__real_stubs_start + (vector_prefetch - __stubs_start)
-		b	__real_stubs_start + (vector_data - __stubs_start)
+		b	__real_stubs_start + (vector_pabt - __stubs_start)
+		b	__real_stubs_start + (vector_dabt - __stubs_start)
 		b	__real_stubs_start + (vector_addrexcptn - __stubs_start)
-		b	__real_stubs_start + (vector_IRQ - __stubs_start)
-		b	__real_stubs_start + (vector_FIQ - __stubs_start)
+		b	__real_stubs_start + (vector_irq - __stubs_start)
+		b	__real_stubs_start + (vector_fiq - __stubs_start)
 
 ENTRY(__trap_init)
 		stmfd	sp!, {r4 - r6, lr}
diff -purN linux-2.6.11-rc2-bk5/arch/arm/mach-pxa/mainstone.c linux-2.6.11-rc2-bk6/arch/arm/mach-pxa/mainstone.c
--- linux-2.6.11-rc2-bk5/arch/arm/mach-pxa/mainstone.c	2005-01-22 02:49:19.000000000 +0100
+++ linux-2.6.11-rc2-bk6/arch/arm/mach-pxa/mainstone.c	2005-01-28 15:50:13.185926833 +0100
@@ -33,6 +33,7 @@
 
 #include <asm/arch/pxa-regs.h>
 #include <asm/arch/mainstone.h>
+#include <asm/arch/audio.h>
 #include <asm/arch/pxafb.h>
 #include <asm/arch/mmc.h>
 
@@ -120,6 +121,44 @@ static struct platform_device smc91x_dev
 	.resource	= smc91x_resources,
 };
 
+static int mst_audio_startup(snd_pcm_substream_t *substream, void *priv)
+{
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		MST_MSCWR2 &= ~MST_MSCWR2_AC97_SPKROFF;
+	return 0;
+}
+
+static void mst_audio_shutdown(snd_pcm_substream_t *substream, void *priv)
+{
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		MST_MSCWR2 |= MST_MSCWR2_AC97_SPKROFF;
+}
+
+static long mst_audio_suspend_mask;
+
+static void mst_audio_suspend(void *priv)
+{
+	mst_audio_suspend_mask = MST_MSCWR2;
+	MST_MSCWR2 |= MST_MSCWR2_AC97_SPKROFF;
+}
+
+static void mst_audio_resume(void *priv)
+{
+	MST_MSCWR2 &= mst_audio_suspend_mask | ~MST_MSCWR2_AC97_SPKROFF;
+}
+
+static pxa2xx_audio_ops_t mst_audio_ops = {
+	.startup	= mst_audio_startup,
+	.shutdown	= mst_audio_shutdown,
+	.suspend	= mst_audio_suspend,
+	.resume		= mst_audio_resume,
+};
+
+static struct platform_device mst_audio_device = {
+	.name		= "pxa2xx-ac97",
+	.id		= -1,
+	.dev		= { .platform_data = &mst_audio_ops },
+};
 
 static void mainstone_backlight_power(int on)
 {
@@ -228,7 +267,14 @@ static struct pxamci_platform_data mains
 
 static void __init mainstone_init(void)
 {
+	/*
+	 * On Mainstone, we route AC97_SYSCLK via GPIO45 to
+	 * the audio daughter card
+	 */
+	pxa_gpio_mode(GPIO45_SYSCLK_AC97_MD);
+
 	platform_device_register(&smc91x_device);
+	platform_device_register(&mst_audio_device);
 
 	/* reading Mainstone's "Virtual Configuration Register"
 	   might be handy to select LCD type here */
diff -purN linux-2.6.11-rc2-bk5/drivers/input/keyboard/atkbd.c linux-2.6.11-rc2-bk6/drivers/input/keyboard/atkbd.c
--- linux-2.6.11-rc2-bk5/drivers/input/keyboard/atkbd.c	2005-01-22 02:49:22.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/input/keyboard/atkbd.c	2005-01-28 15:50:13.509876515 +0100
@@ -234,6 +234,7 @@ static void atkbd_report_key(struct inpu
 	input_regs(dev, regs);
 	if (value == 3) {
 		input_report_key(dev, code, 1);
+		input_sync(dev);
 		input_report_key(dev, code, 0);
 	} else
 		input_event(dev, EV_KEY, code, value);
@@ -352,6 +353,7 @@ static irqreturn_t atkbd_interrupt(struc
 				       "to make it known.\n",
 				       code & 0x80 ? "e0" : "", code & 0x7f);
 			}
+			input_sync(&atkbd->dev);
 			break;
 		case ATKBD_SCR_1:
 			scroll = 1 - atkbd->release * 2;
diff -purN linux-2.6.11-rc2-bk5/drivers/input/serio/i8042.c linux-2.6.11-rc2-bk6/drivers/input/serio/i8042.c
--- linux-2.6.11-rc2-bk5/drivers/input/serio/i8042.c	2005-01-22 02:47:00.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/input/serio/i8042.c	2005-01-28 15:50:13.512876049 +0100
@@ -458,12 +458,11 @@ out:
 }
 
 /*
- * i8042_enable_mux_mode checks whether the controller has an active
- * multiplexor and puts the chip into Multiplexed (as opposed to
- * Legacy) mode.
+ * i8042_set_mux_mode checks whether the controller has an active
+ * multiplexor and puts the chip into Multiplexed (1) or Legacy (0) mode.
  */
 
-static int i8042_enable_mux_mode(struct i8042_values *values, unsigned char *mux_version)
+static int i8042_set_mux_mode(unsigned int mode, unsigned char *mux_version)
 {
 
 	unsigned char param;
@@ -482,11 +481,11 @@ static int i8042_enable_mux_mode(struct 
 	param = 0xf0;
 	if (i8042_command(&param, I8042_CMD_AUX_LOOP) || param != 0x0f)
 		return -1;
-	param = 0x56;
+	param = mode ? 0x56 : 0xf6;
 	if (i8042_command(&param, I8042_CMD_AUX_LOOP) || param != 0xa9)
 		return -1;
-	param = 0xa4;
-	if (i8042_command(&param, I8042_CMD_AUX_LOOP) || param == 0x5b)
+	param = mode ? 0xa4 : 0xa5;
+	if (i8042_command(&param, I8042_CMD_AUX_LOOP) || param == (mode ? 0x5b : 0x5a))
 		return -1;
 
 	if (mux_version)
@@ -540,11 +539,11 @@ static int __init i8042_check_mux(struct
 {
 	unsigned char mux_version;
 
-	if (i8042_enable_mux_mode(values, &mux_version))
+	if (i8042_set_mux_mode(1, &mux_version))
 		return -1;
 
-	/* Workaround for broken chips which seem to support MUX, but in reality don't. */
-	/* They all report version 10.12 */
+	/* Workaround for interference with USB Legacy emulation */
+	/* that causes a v10.12 MUX to be found. */
 	if (mux_version == 0xAC)
 		return -1;
 
@@ -774,12 +773,21 @@ static int i8042_controller_init(void)
  */
 void i8042_controller_reset(void)
 {
-	if (i8042_reset) {
-		unsigned char param;
+	unsigned char param;
+
+/*
+ * Reset the controller if requested.
+ */
 
+	if (i8042_reset)
 		if (i8042_command(&param, I8042_CMD_CTL_TEST))
 			printk(KERN_ERR "i8042.c: i8042 controller reset timeout.\n");
-	}
+
+/*
+ * Disable MUX mode if present.
+ */
+
+	i8042_set_mux_mode(0, NULL);
 
 /*
  * Restore the original control register setting.
@@ -888,7 +896,7 @@ static int i8042_resume(struct device *d
 	}
 
 	if (i8042_mux_present)
-		if (i8042_enable_mux_mode(&i8042_aux_values, NULL) ||
+		if (i8042_set_mux_mode(1, NULL) ||
 		    i8042_enable_mux_ports(&i8042_aux_values)) {
 			printk(KERN_WARNING "i8042: failed to resume active multiplexor, mouse won't work.\n");
 		}
diff -purN linux-2.6.11-rc2-bk5/drivers/input/serio/libps2.c linux-2.6.11-rc2-bk6/drivers/input/serio/libps2.c
--- linux-2.6.11-rc2-bk5/drivers/input/serio/libps2.c	2005-01-22 02:47:30.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/input/serio/libps2.c	2005-01-28 15:50:13.514875738 +0100
@@ -223,7 +223,8 @@ void ps2_init(struct ps2dev *ps2dev, str
 }
 
 /*
- * ps2_handle_ack()
+ * ps2_handle_ack() is supposed to be used in interrupt handler
+ * to properly process ACK/NAK of a command from a PS/2 device.
  */
 
 int ps2_handle_ack(struct ps2dev *ps2dev, unsigned char data)
@@ -250,18 +251,27 @@ int ps2_handle_ack(struct ps2dev *ps2dev
 			}
 			/* Fall through */
 		default:
-			return 1;
+			return 0;
 	}
 
+
 	if (!ps2dev->nak && ps2dev->cmdcnt)
 		ps2dev->flags |= PS2_FLAG_CMD | PS2_FLAG_CMD1;
 
 	ps2dev->flags &= ~PS2_FLAG_ACK;
 	wake_up_interruptible(&ps2dev->wait);
 
-	return data == PS2_RET_ACK || data == PS2_RET_NAK;
+	if (data != PS2_RET_ACK)
+		ps2_handle_response(ps2dev, data);
+
+	return 1;
 }
 
+/*
+ * ps2_handle_response() is supposed to be used in interrupt handler
+ * to properly store device's response to a command and notify process
+ * waiting for completion of the command.
+ */
 
 int ps2_handle_response(struct ps2dev *ps2dev, unsigned char data)
 {
diff -purN linux-2.6.11-rc2-bk5/drivers/net/e100.c linux-2.6.11-rc2-bk6/drivers/net/e100.c
--- linux-2.6.11-rc2-bk5/drivers/net/e100.c	2005-01-22 02:49:19.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/net/e100.c	2005-01-28 15:50:13.819965261 +0100
@@ -1630,6 +1630,7 @@ static void e100_netpoll(struct net_devi
 	struct nic *nic = netdev_priv(netdev);
 	e100_disable_irq(nic);
 	e100_intr(nic->pdev->irq, netdev, NULL);
+	e100_tx_clean(nic);
 	e100_enable_irq(nic);
 }
 #endif
diff -purN linux-2.6.11-rc2-bk5/drivers/net/gianfar.c linux-2.6.11-rc2-bk6/drivers/net/gianfar.c
--- linux-2.6.11-rc2-bk5/drivers/net/gianfar.c	2005-01-22 02:46:42.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/net/gianfar.c	2005-01-28 15:50:13.831963398 +0100
@@ -1190,8 +1190,8 @@ irqreturn_t gfar_receive(int irq, void *
 	} else {
 #ifdef VERBOSE_GFAR_ERRORS
 		printk(KERN_DEBUG "%s: receive called twice (%x)[%x]\n",
-		       dev->name, gfar_read(priv->regs->ievent),
-		       gfar_read(priv->regs->imask));
+		       dev->name, gfar_read(&priv->regs->ievent),
+		       gfar_read(&priv->regs->imask));
 #endif
 	}
 #else
@@ -1415,7 +1415,7 @@ static irqreturn_t gfar_interrupt(int ir
 
 #ifdef VERBOSE_GFAR_ERRORS
 		printk(KERN_DEBUG "%s: busy error (rhalt: %x)\n", dev->name,
-		       gfar_read(priv->regs->rstat));
+		       gfar_read(&priv->regs->rstat));
 #endif
 	}
 	if (events & IEVENT_BABR) {
@@ -1793,7 +1793,7 @@ static irqreturn_t gfar_error(int irq, v
 
 #ifdef VERBOSE_GFAR_ERRORS
 		printk(KERN_DEBUG "%s: busy error (rhalt: %x)\n", dev->name,
-		       gfar_read(priv->regs->rstat));
+		       gfar_read(&priv->regs->rstat));
 #endif
 	}
 	if (events & IEVENT_BABR) {
diff -purN linux-2.6.11-rc2-bk5/drivers/net/skfp/skfddi.c linux-2.6.11-rc2-bk6/drivers/net/skfp/skfddi.c
--- linux-2.6.11-rc2-bk5/drivers/net/skfp/skfddi.c	2005-01-22 02:46:59.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/net/skfp/skfddi.c	2005-01-28 15:50:13.884955167 +0100
@@ -169,8 +169,6 @@ static int num_boards;	/* total number o
 #define PRINTK(s, args...)
 #endif				// DRIVERDEBUG
 
-#define PRIV(dev) (&(((struct s_smc *)dev->priv)->os))
-
 /*
  * =================
  * = skfp_init_one =
@@ -205,7 +203,6 @@ static int skfp_init_one(struct pci_dev 
 {
 	struct net_device *dev;
 	struct s_smc *smc;	/* board pointer */
-	unsigned long port, len;
 	void __iomem *mem;
 	int err;
 
@@ -216,62 +213,43 @@ static int skfp_init_one(struct pci_dev 
 
 	err = pci_enable_device(pdev);
 	if (err)
+		return err;
+
+	err = pci_request_regions(pdev, "skfddi");
+	if (err)
 		goto err_out1;
 
+	pci_set_master(pdev);
 
 #ifdef MEM_MAPPED_IO
 	if (!(pci_resource_flags(pdev, 0) & IORESOURCE_MEM)) {
 		printk(KERN_ERR "skfp: region is not an MMIO resource\n");
 		err = -EIO;
-		goto err_out1;
+		goto err_out2;
 	}
-	port = pci_resource_start(pdev, 0);
-	len = pci_resource_len(pdev, 0);
 
-	if (len < 0x4000) {
-		printk(KERN_ERR "skfp: Invalid PCI region size: %lu\n", len);
-		err = -EIO;
-		goto err_out1;
-	}
+	mem = ioremap(pci_resource_start(pdev, 0), 0x4000);
 #else
 	if (!(pci_resource_flags(pdev, 1) & IO_RESOURCE_IO)) {
 		printk(KERN_ERR "skfp: region is not PIO resource\n");
 		err = -EIO;
-		goto err_out1;
+		goto err_out2;
 	}
 
-	port = pci_resource_start(pdev, 1);
-	len = pci_resource_len(pdev, 1);
-	if (len < FP_IO_LEN) {
-		printk(KERN_ERR "skfp: Invalid PCI region size: %d\n",
-		       io_len);
+	mem = ioport_map(pci_resource_start(pdev, 1), FP_IO_LEN);
+#endif
+	if (!mem) {
+		printk(KERN_ERR "skfp:  Unable to map register, "
+				"FDDI adapter will be disabled.\n");
 		err = -EIO;
-		goto err_out1;
+		goto err_out2;
 	}
-#endif
-	err = pci_request_regions(pdev, "skfddi");
-	if (err)
-		goto err_out1;
-
-	pci_set_master(pdev);
 
 	dev = alloc_fddidev(sizeof(struct s_smc));
 	if (!dev) {
 		printk(KERN_ERR "skfp: Unable to allocate fddi device, "
 				"FDDI adapter will be disabled.\n");
 		err = -ENOMEM;
-		goto err_out2;
-	}
-
-#ifdef MEM_MAPPED_IO
-	mem = ioremap(port, len);
-#else
-	mem =ioport_map(port, len);
-#endif
-	if (!mem) {
-		printk(KERN_ERR "skfp:  Unable to map register, "
-				"FDDI adapter will be disabled.\n");
-		err = -EIO;
 		goto err_out3;
 	}
 
@@ -289,7 +267,7 @@ static int skfp_init_one(struct pci_dev 
 	SET_NETDEV_DEV(dev, &pdev->dev);
 
 	/* Initialize board structure with bus-specific info */
-	smc = (struct s_smc *) dev->priv;
+	smc = netdev_priv(dev);
 	smc->os.dev = dev;
 	smc->os.bus_type = SK_BUS_TYPE_PCI;
 	smc->os.pdev = *pdev;
@@ -331,16 +309,17 @@ err_out5:
 	pci_free_consistent(pdev, MAX_FRAME_SIZE,
 			    smc->os.LocalRxBuffer, smc->os.LocalRxBufferDMA);
 err_out4:
+	free_netdev(dev);
+err_out3:
 #ifdef MEM_MAPPED_IO
-	iounmap(smc->hw.iop);
+	iounmap(mem);
 #else
-	ioport_unmap(smc->hw.iop);
+	ioport_unmap(mem);
 #endif
-err_out3:
-	free_netdev(dev);
 err_out2:
 	pci_release_regions(pdev);
 err_out1:
+	pci_disable_device(pdev);
 	return err;
 }
 
@@ -350,7 +329,7 @@ err_out1:
 static void __devexit skfp_remove_one(struct pci_dev *pdev)
 {
 	struct net_device *p = pci_get_drvdata(pdev);
-	struct s_smc *lp = p->priv;
+	struct s_smc *lp = netdev_priv(p);
 
 	unregister_netdev(p);
 
@@ -376,6 +355,7 @@ static void __devexit skfp_remove_one(st
 	pci_release_regions(pdev);
 	free_netdev(p);
 
+	pci_disable_device(pdev);
 	pci_set_drvdata(pdev, NULL);
 }
 
@@ -406,8 +386,8 @@ static void __devexit skfp_remove_one(st
  */
 static  int skfp_driver_init(struct net_device *dev)
 {
-	struct s_smc *smc = (struct s_smc *) dev->priv;
-	skfddi_priv *bp = PRIV(dev);
+	struct s_smc *smc = netdev_priv(dev);
+	skfddi_priv *bp = &smc->os;
 	int err = -EIO;
 
 	PRINTK(KERN_INFO "entering skfp_driver_init\n");
@@ -513,7 +493,7 @@ fail:
  */
 static int skfp_open(struct net_device *dev)
 {
-	struct s_smc *smc = (struct s_smc *) dev->priv;
+	struct s_smc *smc = netdev_priv(dev);
 	int err;
 
 	PRINTK(KERN_INFO "entering skfp_open\n");
@@ -580,8 +560,8 @@ static int skfp_open(struct net_device *
  */
 static int skfp_close(struct net_device *dev)
 {
-	struct s_smc *smc = (struct s_smc *) dev->priv;
-	skfddi_priv *bp = PRIV(dev);
+	struct s_smc *smc = netdev_priv(dev);
+	skfddi_priv *bp = &smc->os;
 
 	CLI_FBI();
 	smt_reset_defaults(smc, 1);
@@ -640,15 +620,15 @@ irqreturn_t skfp_interrupt(int irq, void
 {
 	struct net_device *dev = (struct net_device *) dev_id;
 	struct s_smc *smc;	/* private board structure pointer */
-	skfddi_priv *bp = PRIV(dev);
-
+	skfddi_priv *bp;
 
 	if (dev == NULL) {
 		printk("%s: irq %d for unknown device\n", dev->name, irq);
 		return IRQ_NONE;
 	}
 
-	smc = (struct s_smc *) dev->priv;
+	smc = netdev_priv(dev);
+	bp = &smc->os;
 
 	// IRQs enabled or disabled ?
 	if (inpd(ADDR(B0_IMSK)) == 0) {
@@ -710,7 +690,7 @@ irqreturn_t skfp_interrupt(int irq, void
  */
 struct net_device_stats *skfp_ctl_get_stats(struct net_device *dev)
 {
-	struct s_smc *bp = (struct s_smc *) dev->priv;
+	struct s_smc *bp = netdev_priv(dev);
 
 	/* Fill the bp->stats structure with driver-maintained counters */
 
@@ -874,7 +854,8 @@ struct net_device_stats *skfp_ctl_get_st
  */
 static void skfp_ctl_set_multicast_list(struct net_device *dev)
 {
-	skfddi_priv *bp = PRIV(dev);
+	struct s_smc *smc = netdev_priv(dev);
+	skfddi_priv *bp = &smc->os;
 	unsigned long Flags;
 
 	spin_lock_irqsave(&bp->DriverLock, Flags);
@@ -887,7 +868,7 @@ static void skfp_ctl_set_multicast_list(
 
 static void skfp_ctl_set_multicast_list_wo_lock(struct net_device *dev)
 {
-	struct s_smc *smc = (struct s_smc *) dev->priv;
+	struct s_smc *smc = netdev_priv(dev);
 	struct dev_mc_list *dmi;	/* ptr to multicast addr entry */
 	int i;
 
@@ -970,9 +951,9 @@ static void skfp_ctl_set_multicast_list_
  */
 static int skfp_ctl_set_mac_address(struct net_device *dev, void *addr)
 {
-	struct s_smc *smc = (struct s_smc *) dev->priv;
+	struct s_smc *smc = netdev_priv(dev);
 	struct sockaddr *p_sockaddr = (struct sockaddr *) addr;
-	skfddi_priv *bp = (skfddi_priv *) & smc->os;
+	skfddi_priv *bp = &smc->os;
 	unsigned long Flags;
 
 
@@ -1010,12 +991,14 @@ static int skfp_ctl_set_mac_address(stru
 
 static int skfp_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
 {
-	skfddi_priv *lp = PRIV(dev);
+	struct s_smc *smc = netdev_priv(dev);
+	skfddi_priv *lp = &smc->os;
 	struct s_skfp_ioctl ioc;
 	int status = 0;
 
 	if (copy_from_user(&ioc, rq->ifr_data, sizeof(struct s_skfp_ioctl)))
 		return -EFAULT;
+
 	switch (ioc.cmd) {
 	case SKFP_GET_STATS:	/* Get the driver statistics */
 		ioc.len = sizeof(lp->MacStat);
@@ -1088,7 +1071,8 @@ static int skfp_ioctl(struct net_device 
  */
 static int skfp_send_pkt(struct sk_buff *skb, struct net_device *dev)
 {
-	skfddi_priv *bp = PRIV(dev);
+	struct s_smc *smc = netdev_priv(dev);
+	skfddi_priv *bp = &smc->os;
 
 	PRINTK(KERN_INFO "skfp_send_pkt\n");
 
@@ -1115,7 +1099,7 @@ static int skfp_send_pkt(struct sk_buff 
 	}
 	bp->QueueSkb--;
 	skb_queue_tail(&bp->SendSkbQueue, skb);
-	send_queued_packets((struct s_smc *) dev->priv);
+	send_queued_packets(netdev_priv(dev));
 	if (bp->QueueSkb == 0) {
 		netif_stop_queue(dev);
 	}
@@ -1150,7 +1134,7 @@ static int skfp_send_pkt(struct sk_buff 
  */
 static void send_queued_packets(struct s_smc *smc)
 {
-	skfddi_priv *bp = (skfddi_priv *) & smc->os;
+	skfddi_priv *bp = &smc->os;
 	struct sk_buff *skb;
 	unsigned char fc;
 	int queue;
@@ -1322,7 +1306,7 @@ static void ResetAdapter(struct s_smc *s
  ************************/
 void llc_restart_tx(struct s_smc *smc)
 {
-	skfddi_priv *bp = (skfddi_priv *) & smc->os;
+	skfddi_priv *bp = &smc->os;
 
 	PRINTK(KERN_INFO "[llc_restart_tx]\n");
 
@@ -1506,7 +1490,7 @@ void dma_complete(struct s_smc *smc, vol
 	 * unmap first, the hardware module could read inconsistent data.
 	 */
 	if (flag & DMA_WR) {
-		skfddi_priv *bp = (skfddi_priv *) & smc->os;
+		skfddi_priv *bp = &smc->os;
 		volatile struct s_smt_fp_rxd *r = &descr->r;
 
 		/* If SKB is NULL, we used the local buffer. */
@@ -1620,7 +1604,7 @@ void dump_data(unsigned char *Data, int 
 void mac_drv_rx_complete(struct s_smc *smc, volatile struct s_smt_fp_rxd *rxd,
 			 int frag_count, int len)
 {
-	skfddi_priv *bp = (skfddi_priv *) & smc->os;
+	skfddi_priv *bp = &smc->os;
 	struct sk_buff *skb;
 	unsigned char *virt, *cp;
 	unsigned short ri;
@@ -1752,7 +1736,7 @@ void mac_drv_requeue_rxd(struct s_smc *s
 
 		printk("fddi: Multi-fragment requeue!\n");
 
-	MaxFrameSize = ((skfddi_priv *) & smc->os)->MaxFrameSize;
+	MaxFrameSize = smc->os.MaxFrameSize;
 	src_rxd = rxd;
 	for (; frag_count > 0; frag_count--) {
 		next_rxd = src_rxd->rxd_next;
@@ -1828,7 +1812,7 @@ void mac_drv_fill_rxd(struct s_smc *smc)
 	// Walk through the list of free receive buffers, passing receive
 	// buffers to the HWM as long as RXDs are available.
 
-	MaxFrameSize = ((skfddi_priv *) & smc->os)->MaxFrameSize;
+	MaxFrameSize = smc->os.MaxFrameSize;
 	// Check if there is any RXD left.
 	while (HWM_GET_RX_FREE(smc) > 0) {
 		PRINTK(KERN_INFO ".\n");
@@ -1897,7 +1881,7 @@ void mac_drv_clear_rxd(struct s_smc *smc
 	for (; frag_count > 0; frag_count--) {
 		skb = rxd->rxd_os.skb;
 		if (skb != NULL) {
-			skfddi_priv *bp = (skfddi_priv *) & smc->os;
+			skfddi_priv *bp = &smc->os;
 			int MaxFrameSize = bp->MaxFrameSize;
 
 			pci_unmap_single(&bp->pdev, rxd->rxd_os.dma_addr,
@@ -1964,7 +1948,7 @@ int mac_drv_rx_init(struct s_smc *smc, i
 	memcpy(skb->data, look_ahead, len);
 
 	// deliver frame to system
-	skb->protocol = fddi_type_trans(skb, ((skfddi_priv *) & smc->os)->dev);
+	skb->protocol = fddi_type_trans(skb, smc->os.dev);
 	skb->dev->last_rx = jiffies;
 	netif_rx(skb);
 
diff -purN linux-2.6.11-rc2-bk5/drivers/net/tulip/de2104x.c linux-2.6.11-rc2-bk6/drivers/net/tulip/de2104x.c
--- linux-2.6.11-rc2-bk5/drivers/net/tulip/de2104x.c	2005-01-22 02:48:45.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/net/tulip/de2104x.c	2005-01-28 15:50:13.936947091 +0100
@@ -1703,6 +1703,7 @@ static void __init de21040_get_mac_addre
 			value = dr32(ROMCmd);
 		while (value < 0 && --boguscnt > 0);
 		de->dev->dev_addr[i] = value;
+		udelay(1);
 		if (boguscnt <= 0)
 			printk(KERN_WARNING PFX "timeout reading 21040 MAC address byte %u\n", i);
 	}
diff -purN linux-2.6.11-rc2-bk5/drivers/scsi/ahci.c linux-2.6.11-rc2-bk6/drivers/scsi/ahci.c
--- linux-2.6.11-rc2-bk5/drivers/scsi/ahci.c	2005-01-22 02:47:30.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/scsi/ahci.c	2005-01-28 15:50:13.986939326 +0100
@@ -246,6 +246,10 @@ static struct pci_device_id ahci_pci_tbl
 	  board_ahci }, /* ICH7 */
 	{ PCI_VENDOR_ID_INTEL, 0x27c5, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
 	  board_ahci }, /* ICH7M */
+	{ PCI_VENDOR_ID_INTEL, 0x27c2, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* ICH7R */
+	{ PCI_VENDOR_ID_INTEL, 0x27c3, PCI_ANY_ID, PCI_ANY_ID, 0, 0,
+	  board_ahci }, /* ICH7R */
 	{ }	/* terminate list */
 };
 
diff -purN linux-2.6.11-rc2-bk5/drivers/scsi/sata_sil.c linux-2.6.11-rc2-bk6/drivers/scsi/sata_sil.c
--- linux-2.6.11-rc2-bk5/drivers/scsi/sata_sil.c	2005-01-22 02:46:42.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/scsi/sata_sil.c	2005-01-28 15:50:14.009935754 +0100
@@ -90,6 +90,7 @@ struct sil_drivelist {
 	{ "ST3120023AS",	SIL_QUIRK_MOD15WRITE },
 	{ "ST3160023AS",	SIL_QUIRK_MOD15WRITE },
 	{ "ST3120026AS",	SIL_QUIRK_MOD15WRITE },
+	{ "ST3200822AS",	SIL_QUIRK_MOD15WRITE },
 	{ "ST340014ASL",	SIL_QUIRK_MOD15WRITE },
 	{ "ST360014ASL",	SIL_QUIRK_MOD15WRITE },
 	{ "ST380011ASL",	SIL_QUIRK_MOD15WRITE },
diff -purN linux-2.6.11-rc2-bk5/drivers/usb/input/hid-core.c linux-2.6.11-rc2-bk6/drivers/usb/input/hid-core.c
--- linux-2.6.11-rc2-bk5/drivers/usb/input/hid-core.c	2005-01-22 02:48:21.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/usb/input/hid-core.c	2005-01-28 15:50:14.039931094 +0100
@@ -925,8 +925,9 @@ static void hid_irq_in(struct urb *urb, 
 			break;
 		case -ECONNRESET:	/* unlink */
 		case -ENOENT:
-		case -ESHUTDOWN:
 		case -EPERM:
+		case -ESHUTDOWN:	/* unplug */
+		case -EILSEQ:		/* unplug timeout on uhci */
 			return;
 		case -ETIMEDOUT:	/* NAK */
 			break;
@@ -1136,12 +1137,15 @@ static void hid_irq_out(struct urb *urb,
 {
 	struct hid_device *hid = urb->context;
 	unsigned long flags;
+	int unplug = 0;
 
 	switch (urb->status) {
 		case 0:			/* success */
+		case -ESHUTDOWN:	/* unplug */
+		case -EILSEQ:		/* unplug timeout on uhci */
+			unplug = 1;
 		case -ECONNRESET:	/* unlink */
 		case -ENOENT:
-		case -ESHUTDOWN:
 			break;
 		default:		/* error */
 			warn("output irq status %d received", urb->status);
@@ -1149,7 +1153,10 @@ static void hid_irq_out(struct urb *urb,
 
 	spin_lock_irqsave(&hid->outlock, flags);
 
-	hid->outtail = (hid->outtail + 1) & (HID_OUTPUT_FIFO_SIZE - 1);
+	if (unplug)
+		hid->outtail = hid->outhead;
+	else
+		hid->outtail = (hid->outtail + 1) & (HID_OUTPUT_FIFO_SIZE - 1);
 
 	if (hid->outhead != hid->outtail) {
 		if (hid_submit_out(hid)) {
@@ -1173,6 +1180,7 @@ static void hid_ctrl(struct urb *urb, st
 {
 	struct hid_device *hid = urb->context;
 	unsigned long flags;
+	int unplug = 0;
 
 	spin_lock_irqsave(&hid->ctrllock, flags);
 
@@ -1180,16 +1188,21 @@ static void hid_ctrl(struct urb *urb, st
 		case 0:			/* success */
 			if (hid->ctrl[hid->ctrltail].dir == USB_DIR_IN) 
 				hid_input_report(hid->ctrl[hid->ctrltail].report->type, urb, regs);
+		case -ESHUTDOWN:	/* unplug */
+		case -EILSEQ:		/* unplug timectrl on uhci */
+			unplug = 1;
 		case -ECONNRESET:	/* unlink */
 		case -ENOENT:
-		case -ESHUTDOWN:
 		case -EPIPE:		/* report not available */
 			break;
 		default:		/* error */
 			warn("ctrl urb status %d received", urb->status);
 	}
 
-	hid->ctrltail = (hid->ctrltail + 1) & (HID_CONTROL_FIFO_SIZE - 1);
+	if (unplug)
+		hid->ctrltail = hid->ctrlhead;
+	else
+		hid->ctrltail = (hid->ctrltail + 1) & (HID_CONTROL_FIFO_SIZE - 1);
 
 	if (hid->ctrlhead != hid->ctrltail) {
 		if (hid_submit_ctrl(hid)) {
diff -purN linux-2.6.11-rc2-bk5/drivers/usb/input/hid-debug.h linux-2.6.11-rc2-bk6/drivers/usb/input/hid-debug.h
--- linux-2.6.11-rc2-bk5/drivers/usb/input/hid-debug.h	2005-01-22 02:48:19.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/usb/input/hid-debug.h	2005-01-28 15:50:14.041930784 +0100
@@ -81,15 +81,21 @@ static const struct hid_usage_entry hid_
       {0, 0x8b, "SystemMenuLeft"},
       {0, 0x8c, "SystemMenuUp"},
       {0, 0x8d, "SystemMenuDown"},
-    {0, 0x90, "D-padUp"},
-    {0, 0x91, "D-padDown"},
-    {0, 0x92, "D-padRight"},
-    {0, 0x93, "D-padLeft"},
+      {0, 0x90, "D-PadUp"},
+      {0, 0x91, "D-PadDown"},
+      {0, 0x92, "D-PadRight"},
+      {0, 0x93, "D-PadLeft"},
   {  7, 0, "Keyboard" },
+      {0, 0x01, "NumLock"},
+      {0, 0x02, "CapsLock"},
+      {0, 0x03, "ScrollLock"},
+      {0, 0x04, "Compose"},
+      {0, 0x05, "Kana"},
   {  8, 0, "LED" },
   {  9, 0, "Button" },
   { 10, 0, "Ordinal" },
-  { 12, 0, "Hotkey" },
+  { 12, 0, "Consumer" },
+      {0, 0x238, "HorizontalWheel"},
   { 13, 0, "Digitizers" },
     {0, 0x01, "Digitizer"},
     {0, 0x02, "Pen"},
@@ -653,13 +659,11 @@ static char *keys[KEY_MAX + 1] = {
 	[KEY_SLOW] = "Slow",			[KEY_SHUFFLE] = "Shuffle",
 	[KEY_BREAK] = "Break",			[KEY_PREVIOUS] = "Previous",
 	[KEY_DIGITS] = "Digits",		[KEY_TEEN] = "TEEN",
-	[KEY_TWEN] = "TWEN",			[KEY_DEL_EOL] = "Delete EOL",
-	[KEY_DEL_EOS] = "Delete EOS",		[KEY_INS_LINE] = "Insert line",
-	[KEY_DEL_LINE] = "Delete line",
+	[KEY_TWEN] = "TWEN",			[KEY_DEL_EOL] = "DeleteEOL",
+	[KEY_DEL_EOS] = "DeleteEOS",		[KEY_INS_LINE] = "InsertLine",
+	[KEY_DEL_LINE] = "DeleteLine",
 };
 
-static char *absval[5] = { "Value", "Min  ", "Max  ", "Fuzz ", "Flat " };
-
 static char *relatives[REL_MAX + 1] = {
 	[0 ... REL_MAX] = NULL,
 	[REL_X] = "X",			[REL_Y] = "Y",
diff -purN linux-2.6.11-rc2-bk5/drivers/usb/input/hid-input.c linux-2.6.11-rc2-bk6/drivers/usb/input/hid-input.c
--- linux-2.6.11-rc2-bk5/drivers/usb/input/hid-input.c	2005-01-22 02:47:49.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/usb/input/hid-input.c	2005-01-28 15:50:14.044930318 +0100
@@ -185,7 +185,9 @@ static void hidinput_configure_usage(str
 			break;
 
 		case HID_UP_LED:
-			map_led((usage->hid - 1) & 0xf);
+			if (usage->hid - 1 >= LED_MAX)
+				goto ignore;
+			map_led(usage->hid - 1);
 			break;
 
 		case HID_UP_DIGITIZER:
@@ -231,7 +233,6 @@ static void hidinput_configure_usage(str
 
 		case HID_UP_CONSUMER:	/* USB HUT v1.1, pages 56-62 */
 
-			set_bit(EV_REP, input->evbit);
 			switch (usage->hid & HID_USAGE) {
 				case 0x000: goto ignore;
 				case 0x034: map_key_clear(KEY_SLEEP);		break;
@@ -268,6 +269,7 @@ static void hidinput_configure_usage(str
 				case 0x226: map_key_clear(KEY_STOP);		break;
 				case 0x227: map_key_clear(KEY_REFRESH);		break;
 				case 0x22a: map_key_clear(KEY_BOOKMARKS);	break;
+				case 0x238: map_rel(REL_HWHEEL);		break;
 				default:    goto unknown;
 			}
 			break;
@@ -288,9 +290,13 @@ static void hidinput_configure_usage(str
 				case 0x084: map_key_clear(KEY_FINANCE);		break;
 				case 0x085: map_key_clear(KEY_SPORT);		break;
 				case 0x086: map_key_clear(KEY_SHOP);	        break;
-				default:    goto unknown;
+				default:    goto ignore;
 			}
 			break;
+
+		case HID_UP_MSVENDOR:
+
+			goto ignore;
 			
 		case HID_UP_PID:
 
@@ -397,11 +403,12 @@ void hidinput_hid_event(struct hid_devic
 	if (!input)
 		return;
 
+	input_regs(input, regs);
+	input_event(input, EV_MSC, MSC_SCAN, usage->hid);
+
 	if (!usage->type)
 		return;
 
-	input_regs(input, regs);
-
 	if (((hid->quirks & HID_QUIRK_2WHEEL_MOUSE_HACK_5) && (usage->hid == 0x00090005))
 		|| ((hid->quirks & HID_QUIRK_2WHEEL_MOUSE_HACK_7) && (usage->hid == 0x00090007))) {
 		if (value) hid->quirks |=  HID_QUIRK_2WHEEL_MOUSE_HACK_ON;
@@ -568,13 +575,16 @@ int hidinput_connect(struct hid_device *
 				hidinput->input.id.product = le16_to_cpu(dev->descriptor.idProduct);
 				hidinput->input.id.version = le16_to_cpu(dev->descriptor.bcdDevice);
 				hidinput->input.dev = &hid->intf->dev;
+
+				set_bit(EV_MSC, hidinput->input.evbit);
+				set_bit(MSC_SCAN, hidinput->input.mscbit);
 			}
 
 			for (i = 0; i < report->maxfield; i++)
 				for (j = 0; j < report->field[i]->maxusage; j++)
 					hidinput_configure_usage(hidinput, report->field[i],
 								 report->field[i]->usage + j);
-
+			
 			if (hid->quirks & HID_QUIRK_MULTI_INPUT) {
 				/* This will leave hidinput NULL, so that it
 				 * allocates another one if we have more inputs on
diff -purN linux-2.6.11-rc2-bk5/drivers/usb/input/hid.h linux-2.6.11-rc2-bk6/drivers/usb/input/hid.h
--- linux-2.6.11-rc2-bk5/drivers/usb/input/hid.h	2005-01-22 02:46:42.000000000 +0100
+++ linux-2.6.11-rc2-bk6/drivers/usb/input/hid.h	2005-01-28 15:50:14.046930007 +0100
@@ -181,6 +181,7 @@ struct hid_item {
 #define HID_UP_DIGITIZER 	0x000d0000
 #define HID_UP_PID 		0x000f0000
 #define HID_UP_HPVENDOR         0xff7f0000
+#define HID_UP_MSVENDOR		0xff000000
 
 #define HID_USAGE		0x0000ffff
 
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-cl7500/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-cl7500/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-cl7500/debug-macro.S	2005-01-28 15:50:06.448906902 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-cl7500/debug-macro.S	2005-01-28 15:50:14.148914166 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-clps711x/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-clps711x/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-clps711x/debug-macro.S	2005-01-28 15:50:06.449906746 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-clps711x/debug-macro.S	2005-01-28 15:50:14.149914011 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-ebsa110/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-ebsa110/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-ebsa110/debug-macro.S	2005-01-28 15:50:06.450906591 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-ebsa110/debug-macro.S	2005-01-28 15:50:14.151913700 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-ebsa285/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-ebsa285/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-ebsa285/debug-macro.S	2005-01-28 15:50:06.452906280 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-ebsa285/debug-macro.S	2005-01-28 15:50:14.152913545 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-epxa10db/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-epxa10db/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-epxa10db/debug-macro.S	2005-01-28 15:50:06.453906125 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-epxa10db/debug-macro.S	2005-01-28 15:50:14.153913390 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-h720x/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-h720x/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-h720x/debug-macro.S	2005-01-28 15:50:06.454905970 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-h720x/debug-macro.S	2005-01-28 15:50:14.155913079 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-imx/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-imx/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-imx/debug-macro.S	2005-01-28 15:50:06.455905815 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-imx/debug-macro.S	2005-01-28 15:50:14.156912924 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-integrator/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-integrator/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-integrator/debug-macro.S	2005-01-28 15:50:06.457905504 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-integrator/debug-macro.S	2005-01-28 15:50:14.157912769 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-iop3xx/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-iop3xx/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-iop3xx/debug-macro.S	2005-01-28 15:50:06.458905349 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-iop3xx/debug-macro.S	2005-01-28 15:50:14.159912458 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-ixp2000/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-ixp2000/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-ixp2000/debug-macro.S	2005-01-28 15:50:06.459905193 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-ixp2000/debug-macro.S	2005-01-28 15:50:14.160912303 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-ixp4xx/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-ixp4xx/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-ixp4xx/debug-macro.S	2005-01-28 15:50:06.461904883 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-ixp4xx/debug-macro.S	2005-01-28 15:50:14.161912147 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-ixp4xx/system.h linux-2.6.11-rc2-bk6/include/asm-arm/arch-ixp4xx/system.h
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-ixp4xx/system.h	2005-01-22 02:48:02.000000000 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-ixp4xx/system.h	2005-01-28 15:50:14.162911992 +0100
@@ -31,13 +31,12 @@ static inline void arch_reset(char mode)
 		/* set the "key" register to enable access to
 		 * "timer" and "enable" registers
 		 */
-		*IXP4XX_OSWK = 0x482e; 	    
+		*IXP4XX_OSWK = IXP4XX_WDT_KEY;
 
-		/* write 0 to the timer register for an immidiate reset */
+		/* write 0 to the timer register for an immediate reset */
 		*IXP4XX_OSWT = 0;
 
-		/* disable watchdog interrupt, enable reset, enable count */
-		*IXP4XX_OSWE = 0x3;
+		*IXP4XX_OSWE = IXP4XX_WDT_RESET_ENABLE | IXP4XX_WDT_COUNT_ENABLE;
 	}
 }
 
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-l7200/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-l7200/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-l7200/debug-macro.S	2005-01-28 15:50:06.462904727 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-l7200/debug-macro.S	2005-01-28 15:50:14.164911681 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-lh7a40x/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-lh7a40x/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-lh7a40x/debug-macro.S	2005-01-28 15:50:06.463904572 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-lh7a40x/debug-macro.S	2005-01-28 15:50:14.165911526 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-omap/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-omap/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-omap/debug-macro.S	2005-01-28 15:50:06.464904417 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-omap/debug-macro.S	2005-01-28 15:50:14.166911371 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-pxa/audio.h linux-2.6.11-rc2-bk6/include/asm-arm/arch-pxa/audio.h
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-pxa/audio.h	1970-01-01 01:00:00.000000000 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-pxa/audio.h	2005-01-28 15:50:14.167911216 +0100
@@ -0,0 +1,16 @@
+#ifndef __ASM_ARCH_AUDIO_H__
+#define __ASM_ARCH_AUDIO_H__
+
+#include <sound/driver.h>
+#include <sound/core.h>
+#include <sound/pcm.h>
+
+typedef struct {
+	int (*startup)(snd_pcm_substream_t *, void *);
+	void (*shutdown)(snd_pcm_substream_t *, void *);
+	void (*suspend)(void *);
+	void (*resume)(void *);
+	void *priv;
+} pxa2xx_audio_ops_t;
+
+#endif
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-pxa/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-pxa/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-pxa/debug-macro.S	2005-01-28 15:50:06.466904106 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-pxa/debug-macro.S	2005-01-28 15:50:14.168911060 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-rpc/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-rpc/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-rpc/debug-macro.S	2005-01-28 15:50:06.467903951 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-rpc/debug-macro.S	2005-01-28 15:50:14.169910905 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-s3c2410/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-s3c2410/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-s3c2410/debug-macro.S	2005-01-28 15:50:06.469903640 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-s3c2410/debug-macro.S	2005-01-28 15:50:14.171910594 +0100
@@ -5,7 +5,7 @@
  *  Copyright (C) 1994-1999 Russell King
  *  Copyright (C) 2005 Simtec Electronics
  *
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-sa1100/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-sa1100/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-sa1100/debug-macro.S	2005-01-28 15:50:06.470903485 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-sa1100/debug-macro.S	2005-01-28 15:50:14.173910284 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-shark/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-shark/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-shark/debug-macro.S	2005-01-28 15:50:06.471903330 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-shark/debug-macro.S	2005-01-28 15:50:14.174910128 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/arch-versatile/debug-macro.S linux-2.6.11-rc2-bk6/include/asm-arm/arch-versatile/debug-macro.S
--- linux-2.6.11-rc2-bk5/include/asm-arm/arch-versatile/debug-macro.S	2005-01-28 15:50:06.472903174 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/arch-versatile/debug-macro.S	2005-01-28 15:50:14.175909973 +0100
@@ -3,7 +3,7 @@
  * Debugging macro include header
  *
  *  Copyright (C) 1994-1999 Russell King
- *  Moved from linux/arch/arm/kernel/debug-armv.S by Ben Dooks
+ *  Moved from linux/arch/arm/kernel/debug.S by Ben Dooks
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
diff -purN linux-2.6.11-rc2-bk5/include/asm-arm/bitops.h linux-2.6.11-rc2-bk6/include/asm-arm/bitops.h
--- linux-2.6.11-rc2-bk5/include/asm-arm/bitops.h	2005-01-22 02:46:58.000000000 +0100
+++ linux-2.6.11-rc2-bk6/include/asm-arm/bitops.h	2005-01-28 15:50:14.177909662 +0100
@@ -224,8 +224,8 @@ extern void _change_bit_be(int nr, volat
 extern int _test_and_set_bit_be(int nr, volatile unsigned long * p);
 extern int _test_and_clear_bit_be(int nr, volatile unsigned long * p);
 extern int _test_and_change_bit_be(int nr, volatile unsigned long * p);
-extern int _find_first_zero_bit_be(void * p, unsigned size);
-extern int _find_next_zero_bit_be(void * p, int size, int offset);
+extern int _find_first_zero_bit_be(const void * p, unsigned size);
+extern int _find_next_zero_bit_be(const void * p, int size, int offset);
 extern int _find_first_bit_be(const unsigned long *p, unsigned size);
 extern int _find_next_bit_be(const unsigned long *p, int size, int offset);
 
